// @aws-sdk/client-cloudfront/dist-cjs/models/models_1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetOriginRequestPolicyConfigRequest = exports.GetOriginRequestPolicyResult = exports.GetOriginRequestPolicyRequest = exports.GetMonitoringSubscriptionResult = exports.GetMonitoringSubscriptionRequest = exports.GetKeyGroupConfigResult = exports.GetKeyGroupConfigRequest = exports.GetKeyGroupResult = exports.GetKeyGroupRequest = exports.NoSuchInvalidation = exports.GetInvalidationResult = exports.GetInvalidationRequest = exports.GetFunctionResult = exports.GetFunctionRequest = exports.GetFieldLevelEncryptionProfileConfigResult = exports.GetFieldLevelEncryptionProfileConfigRequest = exports.GetFieldLevelEncryptionProfileResult = exports.GetFieldLevelEncryptionProfileRequest = exports.GetFieldLevelEncryptionConfigResult = exports.GetFieldLevelEncryptionConfigRequest = exports.GetFieldLevelEncryptionResult = exports.GetFieldLevelEncryptionRequest = exports.GetDistributionConfigResult = exports.GetDistributionConfigRequest = exports.GetDistributionResult = exports.GetDistributionRequest = exports.GetCloudFrontOriginAccessIdentityConfigResult = exports.GetCloudFrontOriginAccessIdentityConfigRequest = exports.GetCloudFrontOriginAccessIdentityResult = exports.GetCloudFrontOriginAccessIdentityRequest = exports.GetCachePolicyConfigResult = exports.GetCachePolicyConfigRequest = exports.GetCachePolicyResult = exports.GetCachePolicyRequest = exports.DescribeFunctionResult = exports.DescribeFunctionRequest = exports.StreamingDistributionNotDisabled = exports.NoSuchStreamingDistribution = exports.DeleteStreamingDistributionRequest = exports.ResponseHeadersPolicyInUse = exports.DeleteResponseHeadersPolicyRequest = exports.RealtimeLogConfigInUse = exports.DeleteRealtimeLogConfigRequest = exports.PublicKeyInUse = exports.DeletePublicKeyRequest = exports.OriginRequestPolicyInUse = exports.DeleteOriginRequestPolicyRequest = exports.DeleteMonitoringSubscriptionResult = exports.DeleteMonitoringSubscriptionRequest = exports.ResourceInUse = void 0;
exports.ListFieldLevelEncryptionProfilesResult = exports.FieldLevelEncryptionProfileList = exports.FieldLevelEncryptionProfileSummary = exports.ListFieldLevelEncryptionProfilesRequest = exports.ListFieldLevelEncryptionConfigsResult = exports.FieldLevelEncryptionList = exports.FieldLevelEncryptionSummary = exports.ListFieldLevelEncryptionConfigsRequest = exports.ListDistributionsByWebACLIdResult = exports.ListDistributionsByWebACLIdRequest = exports.ListDistributionsByResponseHeadersPolicyIdResult = exports.ListDistributionsByResponseHeadersPolicyIdRequest = exports.ListDistributionsByRealtimeLogConfigResult = exports.ListDistributionsByRealtimeLogConfigRequest = exports.ListDistributionsByOriginRequestPolicyIdResult = exports.ListDistributionsByOriginRequestPolicyIdRequest = exports.ListDistributionsByKeyGroupResult = exports.ListDistributionsByKeyGroupRequest = exports.ListDistributionsByCachePolicyIdResult = exports.DistributionIdList = exports.ListDistributionsByCachePolicyIdRequest = exports.ListDistributionsResult = exports.DistributionList = exports.DistributionSummary = exports.ListDistributionsRequest = exports.ListConflictingAliasesResult = exports.ConflictingAliasesList = exports.ConflictingAlias = exports.ListConflictingAliasesRequest = exports.ListCloudFrontOriginAccessIdentitiesResult = exports.CloudFrontOriginAccessIdentityList = exports.CloudFrontOriginAccessIdentitySummary = exports.ListCloudFrontOriginAccessIdentitiesRequest = exports.ListCachePoliciesResult = exports.ListCachePoliciesRequest = exports.GetStreamingDistributionConfigResult = exports.GetStreamingDistributionConfigRequest = exports.GetStreamingDistributionResult = exports.GetStreamingDistributionRequest = exports.GetResponseHeadersPolicyConfigResult = exports.GetResponseHeadersPolicyConfigRequest = exports.GetResponseHeadersPolicyResult = exports.GetResponseHeadersPolicyRequest = exports.GetRealtimeLogConfigResult = exports.GetRealtimeLogConfigRequest = exports.GetPublicKeyConfigResult = exports.GetPublicKeyConfigRequest = exports.GetPublicKeyResult = exports.GetPublicKeyRequest = exports.GetOriginRequestPolicyConfigResult = void 0;
exports.UpdateFieldLevelEncryptionConfigResult = exports.UpdateFieldLevelEncryptionConfigRequest = exports.UpdateDistributionResult = exports.UpdateDistributionRequest = exports.UpdateCloudFrontOriginAccessIdentityResult = exports.UpdateCloudFrontOriginAccessIdentityRequest = exports.UpdateCachePolicyResult = exports.UpdateCachePolicyRequest = exports.UntagResourceRequest = exports.TagKeys = exports.TestFunctionResult = exports.TestResult = exports.TestFunctionRequest = exports.TestFunctionFailed = exports.TagResourceRequest = exports.PublishFunctionResult = exports.PublishFunctionRequest = exports.ListTagsForResourceResult = exports.ListTagsForResourceRequest = exports.ListStreamingDistributionsResult = exports.StreamingDistributionList = exports.StreamingDistributionSummary = exports.ListStreamingDistributionsRequest = exports.ListResponseHeadersPoliciesResult = exports.ResponseHeadersPolicyList = exports.ResponseHeadersPolicySummary = exports.ListResponseHeadersPoliciesRequest = exports.ResponseHeadersPolicyType = exports.ListRealtimeLogConfigsResult = exports.RealtimeLogConfigs = exports.ListRealtimeLogConfigsRequest = exports.ListPublicKeysResult = exports.PublicKeyList = exports.PublicKeySummary = exports.ListPublicKeysRequest = exports.ListOriginRequestPoliciesResult = exports.OriginRequestPolicyList = exports.OriginRequestPolicySummary = exports.ListOriginRequestPoliciesRequest = exports.ListKeyGroupsResult = exports.KeyGroupList = exports.KeyGroupSummary = exports.ListKeyGroupsRequest = exports.ListInvalidationsResult = exports.InvalidationList = exports.InvalidationSummary = exports.ListInvalidationsRequest = exports.ListFunctionsResult = exports.FunctionList = exports.ListFunctionsRequest = void 0;
exports.UpdateStreamingDistributionResult = exports.UpdateStreamingDistributionRequest = exports.UpdateResponseHeadersPolicyResult = exports.UpdateResponseHeadersPolicyRequest = exports.UpdateRealtimeLogConfigResult = exports.UpdateRealtimeLogConfigRequest = exports.UpdatePublicKeyResult = exports.UpdatePublicKeyRequest = exports.UpdateOriginRequestPolicyResult = exports.UpdateOriginRequestPolicyRequest = exports.UpdateKeyGroupResult = exports.UpdateKeyGroupRequest = exports.UpdateFunctionResult = exports.UpdateFunctionRequest = exports.UpdateFieldLevelEncryptionProfileResult = exports.UpdateFieldLevelEncryptionProfileRequest = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const CloudFrontServiceException_1 = require("./CloudFrontServiceException");
const models_0_1 = require("./models_0");
class ResourceInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "ResourceInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.ResourceInUse = ResourceInUse;
var DeleteMonitoringSubscriptionRequest;
(function (DeleteMonitoringSubscriptionRequest) {
    DeleteMonitoringSubscriptionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteMonitoringSubscriptionRequest = exports.DeleteMonitoringSubscriptionRequest || (exports.DeleteMonitoringSubscriptionRequest = {}));
var DeleteMonitoringSubscriptionResult;
(function (DeleteMonitoringSubscriptionResult) {
    DeleteMonitoringSubscriptionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteMonitoringSubscriptionResult = exports.DeleteMonitoringSubscriptionResult || (exports.DeleteMonitoringSubscriptionResult = {}));
var DeleteOriginRequestPolicyRequest;
(function (DeleteOriginRequestPolicyRequest) {
    DeleteOriginRequestPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteOriginRequestPolicyRequest = exports.DeleteOriginRequestPolicyRequest || (exports.DeleteOriginRequestPolicyRequest = {}));
class OriginRequestPolicyInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "OriginRequestPolicyInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "OriginRequestPolicyInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, OriginRequestPolicyInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.OriginRequestPolicyInUse = OriginRequestPolicyInUse;
var DeletePublicKeyRequest;
(function (DeletePublicKeyRequest) {
    DeletePublicKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeletePublicKeyRequest = exports.DeletePublicKeyRequest || (exports.DeletePublicKeyRequest = {}));
class PublicKeyInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "PublicKeyInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "PublicKeyInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, PublicKeyInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.PublicKeyInUse = PublicKeyInUse;
var DeleteRealtimeLogConfigRequest;
(function (DeleteRealtimeLogConfigRequest) {
    DeleteRealtimeLogConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteRealtimeLogConfigRequest = exports.DeleteRealtimeLogConfigRequest || (exports.DeleteRealtimeLogConfigRequest = {}));
class RealtimeLogConfigInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "RealtimeLogConfigInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "RealtimeLogConfigInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, RealtimeLogConfigInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.RealtimeLogConfigInUse = RealtimeLogConfigInUse;
var DeleteResponseHeadersPolicyRequest;
(function (DeleteResponseHeadersPolicyRequest) {
    DeleteResponseHeadersPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteResponseHeadersPolicyRequest = exports.DeleteResponseHeadersPolicyRequest || (exports.DeleteResponseHeadersPolicyRequest = {}));
class ResponseHeadersPolicyInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "ResponseHeadersPolicyInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "ResponseHeadersPolicyInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResponseHeadersPolicyInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.ResponseHeadersPolicyInUse = ResponseHeadersPolicyInUse;
var DeleteStreamingDistributionRequest;
(function (DeleteStreamingDistributionRequest) {
    DeleteStreamingDistributionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteStreamingDistributionRequest = exports.DeleteStreamingDistributionRequest || (exports.DeleteStreamingDistributionRequest = {}));
class NoSuchStreamingDistribution extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchStreamingDistribution",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchStreamingDistribution";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchStreamingDistribution.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchStreamingDistribution = NoSuchStreamingDistribution;
class StreamingDistributionNotDisabled extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "StreamingDistributionNotDisabled",
            $fault: "client",
            ...opts,
        });
        this.name = "StreamingDistributionNotDisabled";
        this.$fault = "client";
        Object.setPrototypeOf(this, StreamingDistributionNotDisabled.prototype);
        this.Message = opts.Message;
    }
}
exports.StreamingDistributionNotDisabled = StreamingDistributionNotDisabled;
var DescribeFunctionRequest;
(function (DescribeFunctionRequest) {
    DescribeFunctionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeFunctionRequest = exports.DescribeFunctionRequest || (exports.DescribeFunctionRequest = {}));
var DescribeFunctionResult;
(function (DescribeFunctionResult) {
    DescribeFunctionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeFunctionResult = exports.DescribeFunctionResult || (exports.DescribeFunctionResult = {}));
var GetCachePolicyRequest;
(function (GetCachePolicyRequest) {
    GetCachePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCachePolicyRequest = exports.GetCachePolicyRequest || (exports.GetCachePolicyRequest = {}));
var GetCachePolicyResult;
(function (GetCachePolicyResult) {
    GetCachePolicyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCachePolicyResult = exports.GetCachePolicyResult || (exports.GetCachePolicyResult = {}));
var GetCachePolicyConfigRequest;
(function (GetCachePolicyConfigRequest) {
    GetCachePolicyConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCachePolicyConfigRequest = exports.GetCachePolicyConfigRequest || (exports.GetCachePolicyConfigRequest = {}));
var GetCachePolicyConfigResult;
(function (GetCachePolicyConfigResult) {
    GetCachePolicyConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCachePolicyConfigResult = exports.GetCachePolicyConfigResult || (exports.GetCachePolicyConfigResult = {}));
var GetCloudFrontOriginAccessIdentityRequest;
(function (GetCloudFrontOriginAccessIdentityRequest) {
    GetCloudFrontOriginAccessIdentityRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCloudFrontOriginAccessIdentityRequest = exports.GetCloudFrontOriginAccessIdentityRequest || (exports.GetCloudFrontOriginAccessIdentityRequest = {}));
var GetCloudFrontOriginAccessIdentityResult;
(function (GetCloudFrontOriginAccessIdentityResult) {
    GetCloudFrontOriginAccessIdentityResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCloudFrontOriginAccessIdentityResult = exports.GetCloudFrontOriginAccessIdentityResult || (exports.GetCloudFrontOriginAccessIdentityResult = {}));
var GetCloudFrontOriginAccessIdentityConfigRequest;
(function (GetCloudFrontOriginAccessIdentityConfigRequest) {
    GetCloudFrontOriginAccessIdentityConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCloudFrontOriginAccessIdentityConfigRequest = exports.GetCloudFrontOriginAccessIdentityConfigRequest || (exports.GetCloudFrontOriginAccessIdentityConfigRequest = {}));
var GetCloudFrontOriginAccessIdentityConfigResult;
(function (GetCloudFrontOriginAccessIdentityConfigResult) {
    GetCloudFrontOriginAccessIdentityConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCloudFrontOriginAccessIdentityConfigResult = exports.GetCloudFrontOriginAccessIdentityConfigResult || (exports.GetCloudFrontOriginAccessIdentityConfigResult = {}));
var GetDistributionRequest;
(function (GetDistributionRequest) {
    GetDistributionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetDistributionRequest = exports.GetDistributionRequest || (exports.GetDistributionRequest = {}));
var GetDistributionResult;
(function (GetDistributionResult) {
    GetDistributionResult.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Distribution && { Distribution: models_0_1.Distribution.filterSensitiveLog(obj.Distribution) }),
    });
})(GetDistributionResult = exports.GetDistributionResult || (exports.GetDistributionResult = {}));
var GetDistributionConfigRequest;
(function (GetDistributionConfigRequest) {
    GetDistributionConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetDistributionConfigRequest = exports.GetDistributionConfigRequest || (exports.GetDistributionConfigRequest = {}));
var GetDistributionConfigResult;
(function (GetDistributionConfigResult) {
    GetDistributionConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.DistributionConfig && {
            DistributionConfig: models_0_1.DistributionConfig.filterSensitiveLog(obj.DistributionConfig),
        }),
    });
})(GetDistributionConfigResult = exports.GetDistributionConfigResult || (exports.GetDistributionConfigResult = {}));
var GetFieldLevelEncryptionRequest;
(function (GetFieldLevelEncryptionRequest) {
    GetFieldLevelEncryptionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFieldLevelEncryptionRequest = exports.GetFieldLevelEncryptionRequest || (exports.GetFieldLevelEncryptionRequest = {}));
var GetFieldLevelEncryptionResult;
(function (GetFieldLevelEncryptionResult) {
    GetFieldLevelEncryptionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFieldLevelEncryptionResult = exports.GetFieldLevelEncryptionResult || (exports.GetFieldLevelEncryptionResult = {}));
var GetFieldLevelEncryptionConfigRequest;
(function (GetFieldLevelEncryptionConfigRequest) {
    GetFieldLevelEncryptionConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFieldLevelEncryptionConfigRequest = exports.GetFieldLevelEncryptionConfigRequest || (exports.GetFieldLevelEncryptionConfigRequest = {}));
var GetFieldLevelEncryptionConfigResult;
(function (GetFieldLevelEncryptionConfigResult) {
    GetFieldLevelEncryptionConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFieldLevelEncryptionConfigResult = exports.GetFieldLevelEncryptionConfigResult || (exports.GetFieldLevelEncryptionConfigResult = {}));
var GetFieldLevelEncryptionProfileRequest;
(function (GetFieldLevelEncryptionProfileRequest) {
    GetFieldLevelEncryptionProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFieldLevelEncryptionProfileRequest = exports.GetFieldLevelEncryptionProfileRequest || (exports.GetFieldLevelEncryptionProfileRequest = {}));
var GetFieldLevelEncryptionProfileResult;
(function (GetFieldLevelEncryptionProfileResult) {
    GetFieldLevelEncryptionProfileResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFieldLevelEncryptionProfileResult = exports.GetFieldLevelEncryptionProfileResult || (exports.GetFieldLevelEncryptionProfileResult = {}));
var GetFieldLevelEncryptionProfileConfigRequest;
(function (GetFieldLevelEncryptionProfileConfigRequest) {
    GetFieldLevelEncryptionProfileConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFieldLevelEncryptionProfileConfigRequest = exports.GetFieldLevelEncryptionProfileConfigRequest || (exports.GetFieldLevelEncryptionProfileConfigRequest = {}));
var GetFieldLevelEncryptionProfileConfigResult;
(function (GetFieldLevelEncryptionProfileConfigResult) {
    GetFieldLevelEncryptionProfileConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFieldLevelEncryptionProfileConfigResult = exports.GetFieldLevelEncryptionProfileConfigResult || (exports.GetFieldLevelEncryptionProfileConfigResult = {}));
var GetFunctionRequest;
(function (GetFunctionRequest) {
    GetFunctionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetFunctionRequest = exports.GetFunctionRequest || (exports.GetFunctionRequest = {}));
var GetFunctionResult;
(function (GetFunctionResult) {
    GetFunctionResult.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.FunctionCode && { FunctionCode: smithy_client_1.SENSITIVE_STRING }),
    });
})(GetFunctionResult = exports.GetFunctionResult || (exports.GetFunctionResult = {}));
var GetInvalidationRequest;
(function (GetInvalidationRequest) {
    GetInvalidationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetInvalidationRequest = exports.GetInvalidationRequest || (exports.GetInvalidationRequest = {}));
var GetInvalidationResult;
(function (GetInvalidationResult) {
    GetInvalidationResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetInvalidationResult = exports.GetInvalidationResult || (exports.GetInvalidationResult = {}));
class NoSuchInvalidation extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchInvalidation",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchInvalidation";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchInvalidation.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchInvalidation = NoSuchInvalidation;
var GetKeyGroupRequest;
(function (GetKeyGroupRequest) {
    GetKeyGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetKeyGroupRequest = exports.GetKeyGroupRequest || (exports.GetKeyGroupRequest = {}));
var GetKeyGroupResult;
(function (GetKeyGroupResult) {
    GetKeyGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetKeyGroupResult = exports.GetKeyGroupResult || (exports.GetKeyGroupResult = {}));
var GetKeyGroupConfigRequest;
(function (GetKeyGroupConfigRequest) {
    GetKeyGroupConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetKeyGroupConfigRequest = exports.GetKeyGroupConfigRequest || (exports.GetKeyGroupConfigRequest = {}));
var GetKeyGroupConfigResult;
(function (GetKeyGroupConfigResult) {
    GetKeyGroupConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetKeyGroupConfigResult = exports.GetKeyGroupConfigResult || (exports.GetKeyGroupConfigResult = {}));
var GetMonitoringSubscriptionRequest;
(function (GetMonitoringSubscriptionRequest) {
    GetMonitoringSubscriptionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetMonitoringSubscriptionRequest = exports.GetMonitoringSubscriptionRequest || (exports.GetMonitoringSubscriptionRequest = {}));
var GetMonitoringSubscriptionResult;
(function (GetMonitoringSubscriptionResult) {
    GetMonitoringSubscriptionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetMonitoringSubscriptionResult = exports.GetMonitoringSubscriptionResult || (exports.GetMonitoringSubscriptionResult = {}));
var GetOriginRequestPolicyRequest;
(function (GetOriginRequestPolicyRequest) {
    GetOriginRequestPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetOriginRequestPolicyRequest = exports.GetOriginRequestPolicyRequest || (exports.GetOriginRequestPolicyRequest = {}));
var GetOriginRequestPolicyResult;
(function (GetOriginRequestPolicyResult) {
    GetOriginRequestPolicyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetOriginRequestPolicyResult = exports.GetOriginRequestPolicyResult || (exports.GetOriginRequestPolicyResult = {}));
var GetOriginRequestPolicyConfigRequest;
(function (GetOriginRequestPolicyConfigRequest) {
    GetOriginRequestPolicyConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetOriginRequestPolicyConfigRequest = exports.GetOriginRequestPolicyConfigRequest || (exports.GetOriginRequestPolicyConfigRequest = {}));
var GetOriginRequestPolicyConfigResult;
(function (GetOriginRequestPolicyConfigResult) {
    GetOriginRequestPolicyConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetOriginRequestPolicyConfigResult = exports.GetOriginRequestPolicyConfigResult || (exports.GetOriginRequestPolicyConfigResult = {}));
var GetPublicKeyRequest;
(function (GetPublicKeyRequest) {
    GetPublicKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPublicKeyRequest = exports.GetPublicKeyRequest || (exports.GetPublicKeyRequest = {}));
var GetPublicKeyResult;
(function (GetPublicKeyResult) {
    GetPublicKeyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPublicKeyResult = exports.GetPublicKeyResult || (exports.GetPublicKeyResult = {}));
var GetPublicKeyConfigRequest;
(function (GetPublicKeyConfigRequest) {
    GetPublicKeyConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPublicKeyConfigRequest = exports.GetPublicKeyConfigRequest || (exports.GetPublicKeyConfigRequest = {}));
var GetPublicKeyConfigResult;
(function (GetPublicKeyConfigResult) {
    GetPublicKeyConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPublicKeyConfigResult = exports.GetPublicKeyConfigResult || (exports.GetPublicKeyConfigResult = {}));
var GetRealtimeLogConfigRequest;
(function (GetRealtimeLogConfigRequest) {
    GetRealtimeLogConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetRealtimeLogConfigRequest = exports.GetRealtimeLogConfigRequest || (exports.GetRealtimeLogConfigRequest = {}));
var GetRealtimeLogConfigResult;
(function (GetRealtimeLogConfigResult) {
    GetRealtimeLogConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetRealtimeLogConfigResult = exports.GetRealtimeLogConfigResult || (exports.GetRealtimeLogConfigResult = {}));
var GetResponseHeadersPolicyRequest;
(function (GetResponseHeadersPolicyRequest) {
    GetResponseHeadersPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetResponseHeadersPolicyRequest = exports.GetResponseHeadersPolicyRequest || (exports.GetResponseHeadersPolicyRequest = {}));
var GetResponseHeadersPolicyResult;
(function (GetResponseHeadersPolicyResult) {
    GetResponseHeadersPolicyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetResponseHeadersPolicyResult = exports.GetResponseHeadersPolicyResult || (exports.GetResponseHeadersPolicyResult = {}));
var GetResponseHeadersPolicyConfigRequest;
(function (GetResponseHeadersPolicyConfigRequest) {
    GetResponseHeadersPolicyConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetResponseHeadersPolicyConfigRequest = exports.GetResponseHeadersPolicyConfigRequest || (exports.GetResponseHeadersPolicyConfigRequest = {}));
var GetResponseHeadersPolicyConfigResult;
(function (GetResponseHeadersPolicyConfigResult) {
    GetResponseHeadersPolicyConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetResponseHeadersPolicyConfigResult = exports.GetResponseHeadersPolicyConfigResult || (exports.GetResponseHeadersPolicyConfigResult = {}));
var GetStreamingDistributionRequest;
(function (GetStreamingDistributionRequest) {
    GetStreamingDistributionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetStreamingDistributionRequest = exports.GetStreamingDistributionRequest || (exports.GetStreamingDistributionRequest = {}));
var GetStreamingDistributionResult;
(function (GetStreamingDistributionResult) {
    GetStreamingDistributionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetStreamingDistributionResult = exports.GetStreamingDistributionResult || (exports.GetStreamingDistributionResult = {}));
var GetStreamingDistributionConfigRequest;
(function (GetStreamingDistributionConfigRequest) {
    GetStreamingDistributionConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetStreamingDistributionConfigRequest = exports.GetStreamingDistributionConfigRequest || (exports.GetStreamingDistributionConfigRequest = {}));
var GetStreamingDistributionConfigResult;
(function (GetStreamingDistributionConfigResult) {
    GetStreamingDistributionConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetStreamingDistributionConfigResult = exports.GetStreamingDistributionConfigResult || (exports.GetStreamingDistributionConfigResult = {}));
var ListCachePoliciesRequest;
(function (ListCachePoliciesRequest) {
    ListCachePoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCachePoliciesRequest = exports.ListCachePoliciesRequest || (exports.ListCachePoliciesRequest = {}));
var ListCachePoliciesResult;
(function (ListCachePoliciesResult) {
    ListCachePoliciesResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCachePoliciesResult = exports.ListCachePoliciesResult || (exports.ListCachePoliciesResult = {}));
var ListCloudFrontOriginAccessIdentitiesRequest;
(function (ListCloudFrontOriginAccessIdentitiesRequest) {
    ListCloudFrontOriginAccessIdentitiesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCloudFrontOriginAccessIdentitiesRequest = exports.ListCloudFrontOriginAccessIdentitiesRequest || (exports.ListCloudFrontOriginAccessIdentitiesRequest = {}));
var CloudFrontOriginAccessIdentitySummary;
(function (CloudFrontOriginAccessIdentitySummary) {
    CloudFrontOriginAccessIdentitySummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CloudFrontOriginAccessIdentitySummary = exports.CloudFrontOriginAccessIdentitySummary || (exports.CloudFrontOriginAccessIdentitySummary = {}));
var CloudFrontOriginAccessIdentityList;
(function (CloudFrontOriginAccessIdentityList) {
    CloudFrontOriginAccessIdentityList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CloudFrontOriginAccessIdentityList = exports.CloudFrontOriginAccessIdentityList || (exports.CloudFrontOriginAccessIdentityList = {}));
var ListCloudFrontOriginAccessIdentitiesResult;
(function (ListCloudFrontOriginAccessIdentitiesResult) {
    ListCloudFrontOriginAccessIdentitiesResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCloudFrontOriginAccessIdentitiesResult = exports.ListCloudFrontOriginAccessIdentitiesResult || (exports.ListCloudFrontOriginAccessIdentitiesResult = {}));
var ListConflictingAliasesRequest;
(function (ListConflictingAliasesRequest) {
    ListConflictingAliasesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListConflictingAliasesRequest = exports.ListConflictingAliasesRequest || (exports.ListConflictingAliasesRequest = {}));
var ConflictingAlias;
(function (ConflictingAlias) {
    ConflictingAlias.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ConflictingAlias = exports.ConflictingAlias || (exports.ConflictingAlias = {}));
var ConflictingAliasesList;
(function (ConflictingAliasesList) {
    ConflictingAliasesList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ConflictingAliasesList = exports.ConflictingAliasesList || (exports.ConflictingAliasesList = {}));
var ListConflictingAliasesResult;
(function (ListConflictingAliasesResult) {
    ListConflictingAliasesResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListConflictingAliasesResult = exports.ListConflictingAliasesResult || (exports.ListConflictingAliasesResult = {}));
var ListDistributionsRequest;
(function (ListDistributionsRequest) {
    ListDistributionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsRequest = exports.ListDistributionsRequest || (exports.ListDistributionsRequest = {}));
var DistributionSummary;
(function (DistributionSummary) {
    DistributionSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DistributionSummary = exports.DistributionSummary || (exports.DistributionSummary = {}));
var DistributionList;
(function (DistributionList) {
    DistributionList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DistributionList = exports.DistributionList || (exports.DistributionList = {}));
var ListDistributionsResult;
(function (ListDistributionsResult) {
    ListDistributionsResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsResult = exports.ListDistributionsResult || (exports.ListDistributionsResult = {}));
var ListDistributionsByCachePolicyIdRequest;
(function (ListDistributionsByCachePolicyIdRequest) {
    ListDistributionsByCachePolicyIdRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByCachePolicyIdRequest = exports.ListDistributionsByCachePolicyIdRequest || (exports.ListDistributionsByCachePolicyIdRequest = {}));
var DistributionIdList;
(function (DistributionIdList) {
    DistributionIdList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DistributionIdList = exports.DistributionIdList || (exports.DistributionIdList = {}));
var ListDistributionsByCachePolicyIdResult;
(function (ListDistributionsByCachePolicyIdResult) {
    ListDistributionsByCachePolicyIdResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByCachePolicyIdResult = exports.ListDistributionsByCachePolicyIdResult || (exports.ListDistributionsByCachePolicyIdResult = {}));
var ListDistributionsByKeyGroupRequest;
(function (ListDistributionsByKeyGroupRequest) {
    ListDistributionsByKeyGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByKeyGroupRequest = exports.ListDistributionsByKeyGroupRequest || (exports.ListDistributionsByKeyGroupRequest = {}));
var ListDistributionsByKeyGroupResult;
(function (ListDistributionsByKeyGroupResult) {
    ListDistributionsByKeyGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByKeyGroupResult = exports.ListDistributionsByKeyGroupResult || (exports.ListDistributionsByKeyGroupResult = {}));
var ListDistributionsByOriginRequestPolicyIdRequest;
(function (ListDistributionsByOriginRequestPolicyIdRequest) {
    ListDistributionsByOriginRequestPolicyIdRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByOriginRequestPolicyIdRequest = exports.ListDistributionsByOriginRequestPolicyIdRequest || (exports.ListDistributionsByOriginRequestPolicyIdRequest = {}));
var ListDistributionsByOriginRequestPolicyIdResult;
(function (ListDistributionsByOriginRequestPolicyIdResult) {
    ListDistributionsByOriginRequestPolicyIdResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByOriginRequestPolicyIdResult = exports.ListDistributionsByOriginRequestPolicyIdResult || (exports.ListDistributionsByOriginRequestPolicyIdResult = {}));
var ListDistributionsByRealtimeLogConfigRequest;
(function (ListDistributionsByRealtimeLogConfigRequest) {
    ListDistributionsByRealtimeLogConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByRealtimeLogConfigRequest = exports.ListDistributionsByRealtimeLogConfigRequest || (exports.ListDistributionsByRealtimeLogConfigRequest = {}));
var ListDistributionsByRealtimeLogConfigResult;
(function (ListDistributionsByRealtimeLogConfigResult) {
    ListDistributionsByRealtimeLogConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByRealtimeLogConfigResult = exports.ListDistributionsByRealtimeLogConfigResult || (exports.ListDistributionsByRealtimeLogConfigResult = {}));
var ListDistributionsByResponseHeadersPolicyIdRequest;
(function (ListDistributionsByResponseHeadersPolicyIdRequest) {
    ListDistributionsByResponseHeadersPolicyIdRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByResponseHeadersPolicyIdRequest = exports.ListDistributionsByResponseHeadersPolicyIdRequest || (exports.ListDistributionsByResponseHeadersPolicyIdRequest = {}));
var ListDistributionsByResponseHeadersPolicyIdResult;
(function (ListDistributionsByResponseHeadersPolicyIdResult) {
    ListDistributionsByResponseHeadersPolicyIdResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByResponseHeadersPolicyIdResult = exports.ListDistributionsByResponseHeadersPolicyIdResult || (exports.ListDistributionsByResponseHeadersPolicyIdResult = {}));
var ListDistributionsByWebACLIdRequest;
(function (ListDistributionsByWebACLIdRequest) {
    ListDistributionsByWebACLIdRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByWebACLIdRequest = exports.ListDistributionsByWebACLIdRequest || (exports.ListDistributionsByWebACLIdRequest = {}));
var ListDistributionsByWebACLIdResult;
(function (ListDistributionsByWebACLIdResult) {
    ListDistributionsByWebACLIdResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDistributionsByWebACLIdResult = exports.ListDistributionsByWebACLIdResult || (exports.ListDistributionsByWebACLIdResult = {}));
var ListFieldLevelEncryptionConfigsRequest;
(function (ListFieldLevelEncryptionConfigsRequest) {
    ListFieldLevelEncryptionConfigsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFieldLevelEncryptionConfigsRequest = exports.ListFieldLevelEncryptionConfigsRequest || (exports.ListFieldLevelEncryptionConfigsRequest = {}));
var FieldLevelEncryptionSummary;
(function (FieldLevelEncryptionSummary) {
    FieldLevelEncryptionSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FieldLevelEncryptionSummary = exports.FieldLevelEncryptionSummary || (exports.FieldLevelEncryptionSummary = {}));
var FieldLevelEncryptionList;
(function (FieldLevelEncryptionList) {
    FieldLevelEncryptionList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FieldLevelEncryptionList = exports.FieldLevelEncryptionList || (exports.FieldLevelEncryptionList = {}));
var ListFieldLevelEncryptionConfigsResult;
(function (ListFieldLevelEncryptionConfigsResult) {
    ListFieldLevelEncryptionConfigsResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFieldLevelEncryptionConfigsResult = exports.ListFieldLevelEncryptionConfigsResult || (exports.ListFieldLevelEncryptionConfigsResult = {}));
var ListFieldLevelEncryptionProfilesRequest;
(function (ListFieldLevelEncryptionProfilesRequest) {
    ListFieldLevelEncryptionProfilesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFieldLevelEncryptionProfilesRequest = exports.ListFieldLevelEncryptionProfilesRequest || (exports.ListFieldLevelEncryptionProfilesRequest = {}));
var FieldLevelEncryptionProfileSummary;
(function (FieldLevelEncryptionProfileSummary) {
    FieldLevelEncryptionProfileSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FieldLevelEncryptionProfileSummary = exports.FieldLevelEncryptionProfileSummary || (exports.FieldLevelEncryptionProfileSummary = {}));
var FieldLevelEncryptionProfileList;
(function (FieldLevelEncryptionProfileList) {
    FieldLevelEncryptionProfileList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FieldLevelEncryptionProfileList = exports.FieldLevelEncryptionProfileList || (exports.FieldLevelEncryptionProfileList = {}));
var ListFieldLevelEncryptionProfilesResult;
(function (ListFieldLevelEncryptionProfilesResult) {
    ListFieldLevelEncryptionProfilesResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFieldLevelEncryptionProfilesResult = exports.ListFieldLevelEncryptionProfilesResult || (exports.ListFieldLevelEncryptionProfilesResult = {}));
var ListFunctionsRequest;
(function (ListFunctionsRequest) {
    ListFunctionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFunctionsRequest = exports.ListFunctionsRequest || (exports.ListFunctionsRequest = {}));
var FunctionList;
(function (FunctionList) {
    FunctionList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FunctionList = exports.FunctionList || (exports.FunctionList = {}));
var ListFunctionsResult;
(function (ListFunctionsResult) {
    ListFunctionsResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFunctionsResult = exports.ListFunctionsResult || (exports.ListFunctionsResult = {}));
var ListInvalidationsRequest;
(function (ListInvalidationsRequest) {
    ListInvalidationsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListInvalidationsRequest = exports.ListInvalidationsRequest || (exports.ListInvalidationsRequest = {}));
var InvalidationSummary;
(function (InvalidationSummary) {
    InvalidationSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidationSummary = exports.InvalidationSummary || (exports.InvalidationSummary = {}));
var InvalidationList;
(function (InvalidationList) {
    InvalidationList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidationList = exports.InvalidationList || (exports.InvalidationList = {}));
var ListInvalidationsResult;
(function (ListInvalidationsResult) {
    ListInvalidationsResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListInvalidationsResult = exports.ListInvalidationsResult || (exports.ListInvalidationsResult = {}));
var ListKeyGroupsRequest;
(function (ListKeyGroupsRequest) {
    ListKeyGroupsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListKeyGroupsRequest = exports.ListKeyGroupsRequest || (exports.ListKeyGroupsRequest = {}));
var KeyGroupSummary;
(function (KeyGroupSummary) {
    KeyGroupSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(KeyGroupSummary = exports.KeyGroupSummary || (exports.KeyGroupSummary = {}));
var KeyGroupList;
(function (KeyGroupList) {
    KeyGroupList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(KeyGroupList = exports.KeyGroupList || (exports.KeyGroupList = {}));
var ListKeyGroupsResult;
(function (ListKeyGroupsResult) {
    ListKeyGroupsResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListKeyGroupsResult = exports.ListKeyGroupsResult || (exports.ListKeyGroupsResult = {}));
var ListOriginRequestPoliciesRequest;
(function (ListOriginRequestPoliciesRequest) {
    ListOriginRequestPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListOriginRequestPoliciesRequest = exports.ListOriginRequestPoliciesRequest || (exports.ListOriginRequestPoliciesRequest = {}));
var OriginRequestPolicySummary;
(function (OriginRequestPolicySummary) {
    OriginRequestPolicySummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginRequestPolicySummary = exports.OriginRequestPolicySummary || (exports.OriginRequestPolicySummary = {}));
var OriginRequestPolicyList;
(function (OriginRequestPolicyList) {
    OriginRequestPolicyList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginRequestPolicyList = exports.OriginRequestPolicyList || (exports.OriginRequestPolicyList = {}));
var ListOriginRequestPoliciesResult;
(function (ListOriginRequestPoliciesResult) {
    ListOriginRequestPoliciesResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListOriginRequestPoliciesResult = exports.ListOriginRequestPoliciesResult || (exports.ListOriginRequestPoliciesResult = {}));
var ListPublicKeysRequest;
(function (ListPublicKeysRequest) {
    ListPublicKeysRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPublicKeysRequest = exports.ListPublicKeysRequest || (exports.ListPublicKeysRequest = {}));
var PublicKeySummary;
(function (PublicKeySummary) {
    PublicKeySummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PublicKeySummary = exports.PublicKeySummary || (exports.PublicKeySummary = {}));
var PublicKeyList;
(function (PublicKeyList) {
    PublicKeyList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PublicKeyList = exports.PublicKeyList || (exports.PublicKeyList = {}));
var ListPublicKeysResult;
(function (ListPublicKeysResult) {
    ListPublicKeysResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPublicKeysResult = exports.ListPublicKeysResult || (exports.ListPublicKeysResult = {}));
var ListRealtimeLogConfigsRequest;
(function (ListRealtimeLogConfigsRequest) {
    ListRealtimeLogConfigsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRealtimeLogConfigsRequest = exports.ListRealtimeLogConfigsRequest || (exports.ListRealtimeLogConfigsRequest = {}));
var RealtimeLogConfigs;
(function (RealtimeLogConfigs) {
    RealtimeLogConfigs.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RealtimeLogConfigs = exports.RealtimeLogConfigs || (exports.RealtimeLogConfigs = {}));
var ListRealtimeLogConfigsResult;
(function (ListRealtimeLogConfigsResult) {
    ListRealtimeLogConfigsResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRealtimeLogConfigsResult = exports.ListRealtimeLogConfigsResult || (exports.ListRealtimeLogConfigsResult = {}));
var ResponseHeadersPolicyType;
(function (ResponseHeadersPolicyType) {
    ResponseHeadersPolicyType["custom"] = "custom";
    ResponseHeadersPolicyType["managed"] = "managed";
})(ResponseHeadersPolicyType = exports.ResponseHeadersPolicyType || (exports.ResponseHeadersPolicyType = {}));
var ListResponseHeadersPoliciesRequest;
(function (ListResponseHeadersPoliciesRequest) {
    ListResponseHeadersPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListResponseHeadersPoliciesRequest = exports.ListResponseHeadersPoliciesRequest || (exports.ListResponseHeadersPoliciesRequest = {}));
var ResponseHeadersPolicySummary;
(function (ResponseHeadersPolicySummary) {
    ResponseHeadersPolicySummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicySummary = exports.ResponseHeadersPolicySummary || (exports.ResponseHeadersPolicySummary = {}));
var ResponseHeadersPolicyList;
(function (ResponseHeadersPolicyList) {
    ResponseHeadersPolicyList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyList = exports.ResponseHeadersPolicyList || (exports.ResponseHeadersPolicyList = {}));
var ListResponseHeadersPoliciesResult;
(function (ListResponseHeadersPoliciesResult) {
    ListResponseHeadersPoliciesResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListResponseHeadersPoliciesResult = exports.ListResponseHeadersPoliciesResult || (exports.ListResponseHeadersPoliciesResult = {}));
var ListStreamingDistributionsRequest;
(function (ListStreamingDistributionsRequest) {
    ListStreamingDistributionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStreamingDistributionsRequest = exports.ListStreamingDistributionsRequest || (exports.ListStreamingDistributionsRequest = {}));
var StreamingDistributionSummary;
(function (StreamingDistributionSummary) {
    StreamingDistributionSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StreamingDistributionSummary = exports.StreamingDistributionSummary || (exports.StreamingDistributionSummary = {}));
var StreamingDistributionList;
(function (StreamingDistributionList) {
    StreamingDistributionList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StreamingDistributionList = exports.StreamingDistributionList || (exports.StreamingDistributionList = {}));
var ListStreamingDistributionsResult;
(function (ListStreamingDistributionsResult) {
    ListStreamingDistributionsResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStreamingDistributionsResult = exports.ListStreamingDistributionsResult || (exports.ListStreamingDistributionsResult = {}));
var ListTagsForResourceRequest;
(function (ListTagsForResourceRequest) {
    ListTagsForResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourceRequest = exports.ListTagsForResourceRequest || (exports.ListTagsForResourceRequest = {}));
var ListTagsForResourceResult;
(function (ListTagsForResourceResult) {
    ListTagsForResourceResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourceResult = exports.ListTagsForResourceResult || (exports.ListTagsForResourceResult = {}));
var PublishFunctionRequest;
(function (PublishFunctionRequest) {
    PublishFunctionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PublishFunctionRequest = exports.PublishFunctionRequest || (exports.PublishFunctionRequest = {}));
var PublishFunctionResult;
(function (PublishFunctionResult) {
    PublishFunctionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PublishFunctionResult = exports.PublishFunctionResult || (exports.PublishFunctionResult = {}));
var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagResourceRequest = exports.TagResourceRequest || (exports.TagResourceRequest = {}));
class TestFunctionFailed extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TestFunctionFailed",
            $fault: "server",
            ...opts,
        });
        this.name = "TestFunctionFailed";
        this.$fault = "server";
        Object.setPrototypeOf(this, TestFunctionFailed.prototype);
        this.Message = opts.Message;
    }
}
exports.TestFunctionFailed = TestFunctionFailed;
var TestFunctionRequest;
(function (TestFunctionRequest) {
    TestFunctionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.EventObject && { EventObject: smithy_client_1.SENSITIVE_STRING }),
    });
})(TestFunctionRequest = exports.TestFunctionRequest || (exports.TestFunctionRequest = {}));
var TestResult;
(function (TestResult) {
    TestResult.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.FunctionExecutionLogs && { FunctionExecutionLogs: smithy_client_1.SENSITIVE_STRING }),
        ...(obj.FunctionErrorMessage && { FunctionErrorMessage: smithy_client_1.SENSITIVE_STRING }),
        ...(obj.FunctionOutput && { FunctionOutput: smithy_client_1.SENSITIVE_STRING }),
    });
})(TestResult = exports.TestResult || (exports.TestResult = {}));
var TestFunctionResult;
(function (TestFunctionResult) {
    TestFunctionResult.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.TestResult && { TestResult: TestResult.filterSensitiveLog(obj.TestResult) }),
    });
})(TestFunctionResult = exports.TestFunctionResult || (exports.TestFunctionResult = {}));
var TagKeys;
(function (TagKeys) {
    TagKeys.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagKeys = exports.TagKeys || (exports.TagKeys = {}));
var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagResourceRequest = exports.UntagResourceRequest || (exports.UntagResourceRequest = {}));
var UpdateCachePolicyRequest;
(function (UpdateCachePolicyRequest) {
    UpdateCachePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateCachePolicyRequest = exports.UpdateCachePolicyRequest || (exports.UpdateCachePolicyRequest = {}));
var UpdateCachePolicyResult;
(function (UpdateCachePolicyResult) {
    UpdateCachePolicyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateCachePolicyResult = exports.UpdateCachePolicyResult || (exports.UpdateCachePolicyResult = {}));
var UpdateCloudFrontOriginAccessIdentityRequest;
(function (UpdateCloudFrontOriginAccessIdentityRequest) {
    UpdateCloudFrontOriginAccessIdentityRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateCloudFrontOriginAccessIdentityRequest = exports.UpdateCloudFrontOriginAccessIdentityRequest || (exports.UpdateCloudFrontOriginAccessIdentityRequest = {}));
var UpdateCloudFrontOriginAccessIdentityResult;
(function (UpdateCloudFrontOriginAccessIdentityResult) {
    UpdateCloudFrontOriginAccessIdentityResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateCloudFrontOriginAccessIdentityResult = exports.UpdateCloudFrontOriginAccessIdentityResult || (exports.UpdateCloudFrontOriginAccessIdentityResult = {}));
var UpdateDistributionRequest;
(function (UpdateDistributionRequest) {
    UpdateDistributionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.DistributionConfig && {
            DistributionConfig: models_0_1.DistributionConfig.filterSensitiveLog(obj.DistributionConfig),
        }),
    });
})(UpdateDistributionRequest = exports.UpdateDistributionRequest || (exports.UpdateDistributionRequest = {}));
var UpdateDistributionResult;
(function (UpdateDistributionResult) {
    UpdateDistributionResult.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Distribution && { Distribution: models_0_1.Distribution.filterSensitiveLog(obj.Distribution) }),
    });
})(UpdateDistributionResult = exports.UpdateDistributionResult || (exports.UpdateDistributionResult = {}));
var UpdateFieldLevelEncryptionConfigRequest;
(function (UpdateFieldLevelEncryptionConfigRequest) {
    UpdateFieldLevelEncryptionConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateFieldLevelEncryptionConfigRequest = exports.UpdateFieldLevelEncryptionConfigRequest || (exports.UpdateFieldLevelEncryptionConfigRequest = {}));
var UpdateFieldLevelEncryptionConfigResult;
(function (UpdateFieldLevelEncryptionConfigResult) {
    UpdateFieldLevelEncryptionConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateFieldLevelEncryptionConfigResult = exports.UpdateFieldLevelEncryptionConfigResult || (exports.UpdateFieldLevelEncryptionConfigResult = {}));
var UpdateFieldLevelEncryptionProfileRequest;
(function (UpdateFieldLevelEncryptionProfileRequest) {
    UpdateFieldLevelEncryptionProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateFieldLevelEncryptionProfileRequest = exports.UpdateFieldLevelEncryptionProfileRequest || (exports.UpdateFieldLevelEncryptionProfileRequest = {}));
var UpdateFieldLevelEncryptionProfileResult;
(function (UpdateFieldLevelEncryptionProfileResult) {
    UpdateFieldLevelEncryptionProfileResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateFieldLevelEncryptionProfileResult = exports.UpdateFieldLevelEncryptionProfileResult || (exports.UpdateFieldLevelEncryptionProfileResult = {}));
var UpdateFunctionRequest;
(function (UpdateFunctionRequest) {
    UpdateFunctionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.FunctionCode && { FunctionCode: smithy_client_1.SENSITIVE_STRING }),
    });
})(UpdateFunctionRequest = exports.UpdateFunctionRequest || (exports.UpdateFunctionRequest = {}));
var UpdateFunctionResult;
(function (UpdateFunctionResult) {
    UpdateFunctionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateFunctionResult = exports.UpdateFunctionResult || (exports.UpdateFunctionResult = {}));
var UpdateKeyGroupRequest;
(function (UpdateKeyGroupRequest) {
    UpdateKeyGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateKeyGroupRequest = exports.UpdateKeyGroupRequest || (exports.UpdateKeyGroupRequest = {}));
var UpdateKeyGroupResult;
(function (UpdateKeyGroupResult) {
    UpdateKeyGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateKeyGroupResult = exports.UpdateKeyGroupResult || (exports.UpdateKeyGroupResult = {}));
var UpdateOriginRequestPolicyRequest;
(function (UpdateOriginRequestPolicyRequest) {
    UpdateOriginRequestPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateOriginRequestPolicyRequest = exports.UpdateOriginRequestPolicyRequest || (exports.UpdateOriginRequestPolicyRequest = {}));
var UpdateOriginRequestPolicyResult;
(function (UpdateOriginRequestPolicyResult) {
    UpdateOriginRequestPolicyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateOriginRequestPolicyResult = exports.UpdateOriginRequestPolicyResult || (exports.UpdateOriginRequestPolicyResult = {}));
var UpdatePublicKeyRequest;
(function (UpdatePublicKeyRequest) {
    UpdatePublicKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdatePublicKeyRequest = exports.UpdatePublicKeyRequest || (exports.UpdatePublicKeyRequest = {}));
var UpdatePublicKeyResult;
(function (UpdatePublicKeyResult) {
    UpdatePublicKeyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdatePublicKeyResult = exports.UpdatePublicKeyResult || (exports.UpdatePublicKeyResult = {}));
var UpdateRealtimeLogConfigRequest;
(function (UpdateRealtimeLogConfigRequest) {
    UpdateRealtimeLogConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateRealtimeLogConfigRequest = exports.UpdateRealtimeLogConfigRequest || (exports.UpdateRealtimeLogConfigRequest = {}));
var UpdateRealtimeLogConfigResult;
(function (UpdateRealtimeLogConfigResult) {
    UpdateRealtimeLogConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateRealtimeLogConfigResult = exports.UpdateRealtimeLogConfigResult || (exports.UpdateRealtimeLogConfigResult = {}));
var UpdateResponseHeadersPolicyRequest;
(function (UpdateResponseHeadersPolicyRequest) {
    UpdateResponseHeadersPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateResponseHeadersPolicyRequest = exports.UpdateResponseHeadersPolicyRequest || (exports.UpdateResponseHeadersPolicyRequest = {}));
var UpdateResponseHeadersPolicyResult;
(function (UpdateResponseHeadersPolicyResult) {
    UpdateResponseHeadersPolicyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateResponseHeadersPolicyResult = exports.UpdateResponseHeadersPolicyResult || (exports.UpdateResponseHeadersPolicyResult = {}));
var UpdateStreamingDistributionRequest;
(function (UpdateStreamingDistributionRequest) {
    UpdateStreamingDistributionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateStreamingDistributionRequest = exports.UpdateStreamingDistributionRequest || (exports.UpdateStreamingDistributionRequest = {}));
var UpdateStreamingDistributionResult;
(function (UpdateStreamingDistributionResult) {
    UpdateStreamingDistributionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateStreamingDistributionResult = exports.UpdateStreamingDistributionResult || (exports.UpdateStreamingDistributionResult = {}));
