// @aws-sdk/client-cloudfront/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudFrontOriginAccessIdentityAlreadyExists = exports.TooManyQueryStringsInCachePolicy = exports.TooManyHeadersInCachePolicy = exports.TooManyCookiesInCachePolicy = exports.TooManyCachePolicies = exports.InconsistentQuantities = exports.CreateCachePolicyResult = exports.CreateCachePolicyRequest = exports.CannotChangeImmutablePublicKeyFields = exports.CachePolicyList = exports.CachePolicySummary = exports.CachePolicyInUse = exports.CachePolicyAlreadyExists = exports.CachePolicy = exports.CachePolicyConfig = exports.ParametersInCacheKeyAndForwardedToOrigin = exports.CachePolicyQueryStringsConfig = exports.QueryStringNames = exports.CachePolicyHeadersConfig = exports.CachePolicyCookiesConfig = exports.CacheBehaviors = exports.CacheBehavior = exports.TrustedSigners = exports.TrustedKeyGroups = exports.LambdaFunctionAssociations = exports.LambdaFunctionAssociation = exports.FunctionAssociations = exports.FunctionAssociation = exports.ForwardedValues = exports.QueryStringCacheKeys = exports.Headers = exports.CookiePreference = exports.CookieNames = exports.BatchTooLarge = exports.TooManyDistributionCNAMEs = exports.NoSuchDistribution = exports.InvalidArgument = exports.IllegalUpdate = exports.AssociateAliasRequest = exports.AllowedMethods = exports.CachedMethods = exports.AliasICPRecordal = exports.Aliases = exports.ActiveTrustedSigners = exports.Signer = exports.ActiveTrustedKeyGroups = exports.KGKeyPairIds = exports.KeyPairIds = exports.AccessDenied = exports.ResponseHeadersPolicyAccessControlAllowMethodsValues = void 0;
exports.InvalidRelativePath = exports.InvalidQueryStringParameters = exports.InvalidProtocolSettings = exports.InvalidOriginReadTimeout = exports.InvalidOriginKeepaliveTimeout = exports.InvalidOriginAccessIdentity = exports.InvalidOrigin = exports.InvalidMinimumProtocolVersion = exports.InvalidLocationCode = exports.InvalidLambdaFunctionAssociation = exports.InvalidHeadersForS3Origin = exports.InvalidGeoRestrictionParameter = exports.InvalidFunctionAssociation = exports.InvalidForwardCookies = exports.InvalidErrorCode = exports.InvalidDefaultRootObject = exports.IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior = exports.DistributionAlreadyExists = exports.CreateDistributionResult = exports.Distribution = exports.CreateDistributionRequest = exports.DistributionConfig = exports.ViewerCertificate = exports.Restrictions = exports.GeoRestriction = exports.Origins = exports.Origin = exports.S3OriginConfig = exports.OriginShield = exports.CustomOriginConfig = exports.OriginSslProtocols = exports.CustomHeaders = exports.OriginCustomHeader = exports.OriginGroups = exports.OriginGroup = exports.OriginGroupMembers = exports.OriginGroupMember = exports.OriginGroupFailoverCriteria = exports.StatusCodes = exports.LoggingConfig = exports.DefaultCacheBehavior = exports.CustomErrorResponses = exports.CustomErrorResponse = exports.CNAMEAlreadyExists = exports.TooManyCloudFrontOriginAccessIdentities = exports.MissingBody = exports.CreateCloudFrontOriginAccessIdentityResult = exports.CloudFrontOriginAccessIdentity = exports.CreateCloudFrontOriginAccessIdentityRequest = exports.CloudFrontOriginAccessIdentityConfig = void 0;
exports.FieldLevelEncryption = exports.CreateFieldLevelEncryptionConfigRequest = exports.FieldLevelEncryptionConfig = exports.QueryArgProfileConfig = exports.QueryArgProfiles = exports.QueryArgProfile = exports.ContentTypeProfileConfig = exports.ContentTypeProfiles = exports.ContentTypeProfile = exports.InvalidTagging = exports.CreateDistributionWithTagsResult = exports.CreateDistributionWithTagsRequest = exports.DistributionConfigWithTags = exports.Tags = exports.Tag = exports.TrustedSignerDoesNotExist = exports.TrustedKeyGroupDoesNotExist = exports.TooManyTrustedSigners = exports.TooManyQueryStringParameters = exports.TooManyOrigins = exports.TooManyOriginGroupsPerDistribution = exports.TooManyOriginCustomHeaders = exports.TooManyLambdaFunctionAssociations = exports.TooManyKeyGroupsAssociatedToDistribution = exports.TooManyHeadersInForwardedValues = exports.TooManyFunctionAssociations = exports.TooManyDistributionsWithSingleFunctionARN = exports.TooManyDistributionsWithLambdaAssociations = exports.TooManyDistributionsWithFunctionAssociations = exports.TooManyDistributionsAssociatedToResponseHeadersPolicy = exports.TooManyDistributionsAssociatedToOriginRequestPolicy = exports.TooManyDistributionsAssociatedToKeyGroup = exports.TooManyDistributionsAssociatedToFieldLevelEncryptionConfig = exports.TooManyDistributionsAssociatedToCachePolicy = exports.TooManyDistributions = exports.TooManyCookieNamesInWhiteList = exports.TooManyCertificates = exports.TooManyCacheBehaviors = exports.RealtimeLogConfigOwnerMismatch = exports.NoSuchResponseHeadersPolicy = exports.NoSuchRealtimeLogConfig = exports.NoSuchOriginRequestPolicy = exports.NoSuchOrigin = exports.NoSuchFieldLevelEncryptionConfig = exports.NoSuchCachePolicy = exports.InvalidWebACLId = exports.InvalidViewerCertificate = exports.InvalidTTLOrder = exports.InvalidResponseCode = exports.InvalidRequiredProtocol = void 0;
exports.OriginRequestPolicyCookiesConfig = exports.CreateMonitoringSubscriptionResult = exports.CreateMonitoringSubscriptionRequest = exports.MonitoringSubscription = exports.RealtimeMetricsSubscriptionConfig = exports.RealtimeMetricsSubscriptionStatus = exports.TooManyPublicKeysInKeyGroup = exports.TooManyKeyGroups = exports.KeyGroupAlreadyExists = exports.CreateKeyGroupResult = exports.KeyGroup = exports.CreateKeyGroupRequest = exports.KeyGroupConfig = exports.TooManyInvalidationsInProgress = exports.CreateInvalidationResult = exports.Invalidation = exports.CreateInvalidationRequest = exports.InvalidationBatch = exports.Paths = exports.UnsupportedOperation = exports.TooManyFunctions = exports.FunctionSizeLimitExceeded = exports.FunctionAlreadyExists = exports.CreateFunctionResult = exports.FunctionSummary = exports.FunctionMetadata = exports.FunctionStage = exports.CreateFunctionRequest = exports.FunctionConfig = exports.FunctionRuntime = exports.TooManyFieldLevelEncryptionProfiles = exports.TooManyFieldLevelEncryptionFieldPatterns = exports.TooManyFieldLevelEncryptionEncryptionEntities = exports.NoSuchPublicKey = exports.FieldLevelEncryptionProfileSizeExceeded = exports.FieldLevelEncryptionProfileAlreadyExists = exports.CreateFieldLevelEncryptionProfileResult = exports.FieldLevelEncryptionProfile = exports.CreateFieldLevelEncryptionProfileRequest = exports.FieldLevelEncryptionProfileConfig = exports.EncryptionEntities = exports.EncryptionEntity = exports.FieldPatterns = exports.TooManyFieldLevelEncryptionQueryArgProfiles = exports.TooManyFieldLevelEncryptionContentTypeProfiles = exports.TooManyFieldLevelEncryptionConfigs = exports.QueryArgProfileEmpty = exports.NoSuchFieldLevelEncryptionProfile = exports.FieldLevelEncryptionConfigAlreadyExists = exports.CreateFieldLevelEncryptionConfigResult = void 0;
exports.StreamingLoggingConfig = exports.TooManyResponseHeadersPolicies = exports.TooManyCustomHeadersInResponseHeadersPolicy = exports.TooLongCSPInResponseHeadersPolicy = exports.ResponseHeadersPolicyAlreadyExists = exports.CreateResponseHeadersPolicyResult = exports.ResponseHeadersPolicy = exports.CreateResponseHeadersPolicyRequest = exports.ResponseHeadersPolicyConfig = exports.ResponseHeadersPolicyServerTimingHeadersConfig = exports.ResponseHeadersPolicySecurityHeadersConfig = exports.ResponseHeadersPolicyXSSProtection = exports.ResponseHeadersPolicyStrictTransportSecurity = exports.ResponseHeadersPolicyReferrerPolicy = exports.ReferrerPolicyList = exports.ResponseHeadersPolicyFrameOptions = exports.FrameOptionsList = exports.ResponseHeadersPolicyContentTypeOptions = exports.ResponseHeadersPolicyContentSecurityPolicy = exports.ResponseHeadersPolicyCustomHeadersConfig = exports.ResponseHeadersPolicyCustomHeader = exports.ResponseHeadersPolicyCorsConfig = exports.ResponseHeadersPolicyAccessControlExposeHeaders = exports.ResponseHeadersPolicyAccessControlAllowOrigins = exports.ResponseHeadersPolicyAccessControlAllowMethods = exports.ResponseHeadersPolicyAccessControlAllowHeaders = exports.TooManyRealtimeLogConfigs = exports.RealtimeLogConfigAlreadyExists = exports.CreateRealtimeLogConfigResult = exports.RealtimeLogConfig = exports.CreateRealtimeLogConfigRequest = exports.EndPoint = exports.KinesisStreamConfig = exports.TooManyPublicKeys = exports.PublicKeyAlreadyExists = exports.CreatePublicKeyResult = exports.PublicKey = exports.CreatePublicKeyRequest = exports.PublicKeyConfig = exports.TooManyQueryStringsInOriginRequestPolicy = exports.TooManyOriginRequestPolicies = exports.TooManyHeadersInOriginRequestPolicy = exports.TooManyCookiesInOriginRequestPolicy = exports.OriginRequestPolicyAlreadyExists = exports.CreateOriginRequestPolicyResult = exports.OriginRequestPolicy = exports.CreateOriginRequestPolicyRequest = exports.OriginRequestPolicyConfig = exports.OriginRequestPolicyQueryStringsConfig = exports.OriginRequestPolicyHeadersConfig = void 0;
exports.NoSuchResource = exports.DeleteKeyGroupRequest = exports.NoSuchFunctionExists = exports.FunctionInUse = exports.DeleteFunctionRequest = exports.FieldLevelEncryptionProfileInUse = exports.DeleteFieldLevelEncryptionProfileRequest = exports.FieldLevelEncryptionConfigInUse = exports.DeleteFieldLevelEncryptionConfigRequest = exports.DistributionNotDisabled = exports.DeleteDistributionRequest = exports.NoSuchCloudFrontOriginAccessIdentity = exports.DeleteCloudFrontOriginAccessIdentityRequest = exports.CloudFrontOriginAccessIdentityInUse = exports.PreconditionFailed = exports.InvalidIfMatchVersion = exports.IllegalDelete = exports.DeleteCachePolicyRequest = exports.CreateStreamingDistributionWithTagsResult = exports.CreateStreamingDistributionWithTagsRequest = exports.StreamingDistributionConfigWithTags = exports.TooManyStreamingDistributions = exports.TooManyStreamingDistributionCNAMEs = exports.StreamingDistributionAlreadyExists = exports.CreateStreamingDistributionResult = exports.StreamingDistribution = exports.CreateStreamingDistributionRequest = exports.StreamingDistributionConfig = exports.S3Origin = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const CloudFrontServiceException_1 = require("./CloudFrontServiceException");
var ResponseHeadersPolicyAccessControlAllowMethodsValues;
(function (ResponseHeadersPolicyAccessControlAllowMethodsValues) {
    ResponseHeadersPolicyAccessControlAllowMethodsValues["ALL"] = "ALL";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["DELETE"] = "DELETE";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["GET"] = "GET";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["HEAD"] = "HEAD";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["OPTIONS"] = "OPTIONS";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["PATCH"] = "PATCH";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["POST"] = "POST";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["PUT"] = "PUT";
})(ResponseHeadersPolicyAccessControlAllowMethodsValues = exports.ResponseHeadersPolicyAccessControlAllowMethodsValues || (exports.ResponseHeadersPolicyAccessControlAllowMethodsValues = {}));
class AccessDenied extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "AccessDenied",
            $fault: "client",
            ...opts,
        });
        this.name = "AccessDenied";
        this.$fault = "client";
        Object.setPrototypeOf(this, AccessDenied.prototype);
        this.Message = opts.Message;
    }
}
exports.AccessDenied = AccessDenied;
var KeyPairIds;
(function (KeyPairIds) {
    KeyPairIds.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(KeyPairIds = exports.KeyPairIds || (exports.KeyPairIds = {}));
var KGKeyPairIds;
(function (KGKeyPairIds) {
    KGKeyPairIds.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(KGKeyPairIds = exports.KGKeyPairIds || (exports.KGKeyPairIds = {}));
var ActiveTrustedKeyGroups;
(function (ActiveTrustedKeyGroups) {
    ActiveTrustedKeyGroups.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ActiveTrustedKeyGroups = exports.ActiveTrustedKeyGroups || (exports.ActiveTrustedKeyGroups = {}));
var Signer;
(function (Signer) {
    Signer.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Signer = exports.Signer || (exports.Signer = {}));
var ActiveTrustedSigners;
(function (ActiveTrustedSigners) {
    ActiveTrustedSigners.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ActiveTrustedSigners = exports.ActiveTrustedSigners || (exports.ActiveTrustedSigners = {}));
var Aliases;
(function (Aliases) {
    Aliases.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Aliases = exports.Aliases || (exports.Aliases = {}));
var AliasICPRecordal;
(function (AliasICPRecordal) {
    AliasICPRecordal.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AliasICPRecordal = exports.AliasICPRecordal || (exports.AliasICPRecordal = {}));
var CachedMethods;
(function (CachedMethods) {
    CachedMethods.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CachedMethods = exports.CachedMethods || (exports.CachedMethods = {}));
var AllowedMethods;
(function (AllowedMethods) {
    AllowedMethods.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AllowedMethods = exports.AllowedMethods || (exports.AllowedMethods = {}));
var AssociateAliasRequest;
(function (AssociateAliasRequest) {
    AssociateAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AssociateAliasRequest = exports.AssociateAliasRequest || (exports.AssociateAliasRequest = {}));
class IllegalUpdate extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "IllegalUpdate",
            $fault: "client",
            ...opts,
        });
        this.name = "IllegalUpdate";
        this.$fault = "client";
        Object.setPrototypeOf(this, IllegalUpdate.prototype);
        this.Message = opts.Message;
    }
}
exports.IllegalUpdate = IllegalUpdate;
class InvalidArgument extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidArgument",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidArgument";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidArgument.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidArgument = InvalidArgument;
class NoSuchDistribution extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchDistribution",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchDistribution";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchDistribution.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchDistribution = NoSuchDistribution;
class TooManyDistributionCNAMEs extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyDistributionCNAMEs",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionCNAMEs";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionCNAMEs.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionCNAMEs = TooManyDistributionCNAMEs;
class BatchTooLarge extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "BatchTooLarge",
            $fault: "client",
            ...opts,
        });
        this.name = "BatchTooLarge";
        this.$fault = "client";
        Object.setPrototypeOf(this, BatchTooLarge.prototype);
        this.Message = opts.Message;
    }
}
exports.BatchTooLarge = BatchTooLarge;
var CookieNames;
(function (CookieNames) {
    CookieNames.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CookieNames = exports.CookieNames || (exports.CookieNames = {}));
var CookiePreference;
(function (CookiePreference) {
    CookiePreference.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CookiePreference = exports.CookiePreference || (exports.CookiePreference = {}));
var Headers;
(function (Headers) {
    Headers.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Headers = exports.Headers || (exports.Headers = {}));
var QueryStringCacheKeys;
(function (QueryStringCacheKeys) {
    QueryStringCacheKeys.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(QueryStringCacheKeys = exports.QueryStringCacheKeys || (exports.QueryStringCacheKeys = {}));
var ForwardedValues;
(function (ForwardedValues) {
    ForwardedValues.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ForwardedValues = exports.ForwardedValues || (exports.ForwardedValues = {}));
var FunctionAssociation;
(function (FunctionAssociation) {
    FunctionAssociation.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FunctionAssociation = exports.FunctionAssociation || (exports.FunctionAssociation = {}));
var FunctionAssociations;
(function (FunctionAssociations) {
    FunctionAssociations.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FunctionAssociations = exports.FunctionAssociations || (exports.FunctionAssociations = {}));
var LambdaFunctionAssociation;
(function (LambdaFunctionAssociation) {
    LambdaFunctionAssociation.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LambdaFunctionAssociation = exports.LambdaFunctionAssociation || (exports.LambdaFunctionAssociation = {}));
var LambdaFunctionAssociations;
(function (LambdaFunctionAssociations) {
    LambdaFunctionAssociations.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LambdaFunctionAssociations = exports.LambdaFunctionAssociations || (exports.LambdaFunctionAssociations = {}));
var TrustedKeyGroups;
(function (TrustedKeyGroups) {
    TrustedKeyGroups.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TrustedKeyGroups = exports.TrustedKeyGroups || (exports.TrustedKeyGroups = {}));
var TrustedSigners;
(function (TrustedSigners) {
    TrustedSigners.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TrustedSigners = exports.TrustedSigners || (exports.TrustedSigners = {}));
var CacheBehavior;
(function (CacheBehavior) {
    CacheBehavior.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CacheBehavior = exports.CacheBehavior || (exports.CacheBehavior = {}));
var CacheBehaviors;
(function (CacheBehaviors) {
    CacheBehaviors.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CacheBehaviors = exports.CacheBehaviors || (exports.CacheBehaviors = {}));
var CachePolicyCookiesConfig;
(function (CachePolicyCookiesConfig) {
    CachePolicyCookiesConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CachePolicyCookiesConfig = exports.CachePolicyCookiesConfig || (exports.CachePolicyCookiesConfig = {}));
var CachePolicyHeadersConfig;
(function (CachePolicyHeadersConfig) {
    CachePolicyHeadersConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CachePolicyHeadersConfig = exports.CachePolicyHeadersConfig || (exports.CachePolicyHeadersConfig = {}));
var QueryStringNames;
(function (QueryStringNames) {
    QueryStringNames.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(QueryStringNames = exports.QueryStringNames || (exports.QueryStringNames = {}));
var CachePolicyQueryStringsConfig;
(function (CachePolicyQueryStringsConfig) {
    CachePolicyQueryStringsConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CachePolicyQueryStringsConfig = exports.CachePolicyQueryStringsConfig || (exports.CachePolicyQueryStringsConfig = {}));
var ParametersInCacheKeyAndForwardedToOrigin;
(function (ParametersInCacheKeyAndForwardedToOrigin) {
    ParametersInCacheKeyAndForwardedToOrigin.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ParametersInCacheKeyAndForwardedToOrigin = exports.ParametersInCacheKeyAndForwardedToOrigin || (exports.ParametersInCacheKeyAndForwardedToOrigin = {}));
var CachePolicyConfig;
(function (CachePolicyConfig) {
    CachePolicyConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CachePolicyConfig = exports.CachePolicyConfig || (exports.CachePolicyConfig = {}));
var CachePolicy;
(function (CachePolicy) {
    CachePolicy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CachePolicy = exports.CachePolicy || (exports.CachePolicy = {}));
class CachePolicyAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "CachePolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "CachePolicyAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, CachePolicyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.CachePolicyAlreadyExists = CachePolicyAlreadyExists;
class CachePolicyInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "CachePolicyInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "CachePolicyInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, CachePolicyInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.CachePolicyInUse = CachePolicyInUse;
var CachePolicySummary;
(function (CachePolicySummary) {
    CachePolicySummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CachePolicySummary = exports.CachePolicySummary || (exports.CachePolicySummary = {}));
var CachePolicyList;
(function (CachePolicyList) {
    CachePolicyList.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CachePolicyList = exports.CachePolicyList || (exports.CachePolicyList = {}));
class CannotChangeImmutablePublicKeyFields extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "CannotChangeImmutablePublicKeyFields",
            $fault: "client",
            ...opts,
        });
        this.name = "CannotChangeImmutablePublicKeyFields";
        this.$fault = "client";
        Object.setPrototypeOf(this, CannotChangeImmutablePublicKeyFields.prototype);
        this.Message = opts.Message;
    }
}
exports.CannotChangeImmutablePublicKeyFields = CannotChangeImmutablePublicKeyFields;
var CreateCachePolicyRequest;
(function (CreateCachePolicyRequest) {
    CreateCachePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateCachePolicyRequest = exports.CreateCachePolicyRequest || (exports.CreateCachePolicyRequest = {}));
var CreateCachePolicyResult;
(function (CreateCachePolicyResult) {
    CreateCachePolicyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateCachePolicyResult = exports.CreateCachePolicyResult || (exports.CreateCachePolicyResult = {}));
class InconsistentQuantities extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InconsistentQuantities",
            $fault: "client",
            ...opts,
        });
        this.name = "InconsistentQuantities";
        this.$fault = "client";
        Object.setPrototypeOf(this, InconsistentQuantities.prototype);
        this.Message = opts.Message;
    }
}
exports.InconsistentQuantities = InconsistentQuantities;
class TooManyCachePolicies extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyCachePolicies",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCachePolicies";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCachePolicies.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCachePolicies = TooManyCachePolicies;
class TooManyCookiesInCachePolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyCookiesInCachePolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCookiesInCachePolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCookiesInCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCookiesInCachePolicy = TooManyCookiesInCachePolicy;
class TooManyHeadersInCachePolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyHeadersInCachePolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyHeadersInCachePolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyHeadersInCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyHeadersInCachePolicy = TooManyHeadersInCachePolicy;
class TooManyQueryStringsInCachePolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyQueryStringsInCachePolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyQueryStringsInCachePolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyQueryStringsInCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyQueryStringsInCachePolicy = TooManyQueryStringsInCachePolicy;
class CloudFrontOriginAccessIdentityAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "CloudFrontOriginAccessIdentityAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "CloudFrontOriginAccessIdentityAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, CloudFrontOriginAccessIdentityAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.CloudFrontOriginAccessIdentityAlreadyExists = CloudFrontOriginAccessIdentityAlreadyExists;
var CloudFrontOriginAccessIdentityConfig;
(function (CloudFrontOriginAccessIdentityConfig) {
    CloudFrontOriginAccessIdentityConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CloudFrontOriginAccessIdentityConfig = exports.CloudFrontOriginAccessIdentityConfig || (exports.CloudFrontOriginAccessIdentityConfig = {}));
var CreateCloudFrontOriginAccessIdentityRequest;
(function (CreateCloudFrontOriginAccessIdentityRequest) {
    CreateCloudFrontOriginAccessIdentityRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateCloudFrontOriginAccessIdentityRequest = exports.CreateCloudFrontOriginAccessIdentityRequest || (exports.CreateCloudFrontOriginAccessIdentityRequest = {}));
var CloudFrontOriginAccessIdentity;
(function (CloudFrontOriginAccessIdentity) {
    CloudFrontOriginAccessIdentity.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CloudFrontOriginAccessIdentity = exports.CloudFrontOriginAccessIdentity || (exports.CloudFrontOriginAccessIdentity = {}));
var CreateCloudFrontOriginAccessIdentityResult;
(function (CreateCloudFrontOriginAccessIdentityResult) {
    CreateCloudFrontOriginAccessIdentityResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateCloudFrontOriginAccessIdentityResult = exports.CreateCloudFrontOriginAccessIdentityResult || (exports.CreateCloudFrontOriginAccessIdentityResult = {}));
class MissingBody extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "MissingBody",
            $fault: "client",
            ...opts,
        });
        this.name = "MissingBody";
        this.$fault = "client";
        Object.setPrototypeOf(this, MissingBody.prototype);
        this.Message = opts.Message;
    }
}
exports.MissingBody = MissingBody;
class TooManyCloudFrontOriginAccessIdentities extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyCloudFrontOriginAccessIdentities",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCloudFrontOriginAccessIdentities";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCloudFrontOriginAccessIdentities.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCloudFrontOriginAccessIdentities = TooManyCloudFrontOriginAccessIdentities;
class CNAMEAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "CNAMEAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "CNAMEAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, CNAMEAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.CNAMEAlreadyExists = CNAMEAlreadyExists;
var CustomErrorResponse;
(function (CustomErrorResponse) {
    CustomErrorResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CustomErrorResponse = exports.CustomErrorResponse || (exports.CustomErrorResponse = {}));
var CustomErrorResponses;
(function (CustomErrorResponses) {
    CustomErrorResponses.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CustomErrorResponses = exports.CustomErrorResponses || (exports.CustomErrorResponses = {}));
var DefaultCacheBehavior;
(function (DefaultCacheBehavior) {
    DefaultCacheBehavior.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DefaultCacheBehavior = exports.DefaultCacheBehavior || (exports.DefaultCacheBehavior = {}));
var LoggingConfig;
(function (LoggingConfig) {
    LoggingConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LoggingConfig = exports.LoggingConfig || (exports.LoggingConfig = {}));
var StatusCodes;
(function (StatusCodes) {
    StatusCodes.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StatusCodes = exports.StatusCodes || (exports.StatusCodes = {}));
var OriginGroupFailoverCriteria;
(function (OriginGroupFailoverCriteria) {
    OriginGroupFailoverCriteria.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginGroupFailoverCriteria = exports.OriginGroupFailoverCriteria || (exports.OriginGroupFailoverCriteria = {}));
var OriginGroupMember;
(function (OriginGroupMember) {
    OriginGroupMember.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginGroupMember = exports.OriginGroupMember || (exports.OriginGroupMember = {}));
var OriginGroupMembers;
(function (OriginGroupMembers) {
    OriginGroupMembers.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginGroupMembers = exports.OriginGroupMembers || (exports.OriginGroupMembers = {}));
var OriginGroup;
(function (OriginGroup) {
    OriginGroup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginGroup = exports.OriginGroup || (exports.OriginGroup = {}));
var OriginGroups;
(function (OriginGroups) {
    OriginGroups.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginGroups = exports.OriginGroups || (exports.OriginGroups = {}));
var OriginCustomHeader;
(function (OriginCustomHeader) {
    OriginCustomHeader.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.HeaderValue && { HeaderValue: smithy_client_1.SENSITIVE_STRING }),
    });
})(OriginCustomHeader = exports.OriginCustomHeader || (exports.OriginCustomHeader = {}));
var CustomHeaders;
(function (CustomHeaders) {
    CustomHeaders.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Items && { Items: obj.Items.map((item) => OriginCustomHeader.filterSensitiveLog(item)) }),
    });
})(CustomHeaders = exports.CustomHeaders || (exports.CustomHeaders = {}));
var OriginSslProtocols;
(function (OriginSslProtocols) {
    OriginSslProtocols.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginSslProtocols = exports.OriginSslProtocols || (exports.OriginSslProtocols = {}));
var CustomOriginConfig;
(function (CustomOriginConfig) {
    CustomOriginConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CustomOriginConfig = exports.CustomOriginConfig || (exports.CustomOriginConfig = {}));
var OriginShield;
(function (OriginShield) {
    OriginShield.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginShield = exports.OriginShield || (exports.OriginShield = {}));
var S3OriginConfig;
(function (S3OriginConfig) {
    S3OriginConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(S3OriginConfig = exports.S3OriginConfig || (exports.S3OriginConfig = {}));
var Origin;
(function (Origin) {
    Origin.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.CustomHeaders && { CustomHeaders: CustomHeaders.filterSensitiveLog(obj.CustomHeaders) }),
    });
})(Origin = exports.Origin || (exports.Origin = {}));
var Origins;
(function (Origins) {
    Origins.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Origins = exports.Origins || (exports.Origins = {}));
var GeoRestriction;
(function (GeoRestriction) {
    GeoRestriction.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GeoRestriction = exports.GeoRestriction || (exports.GeoRestriction = {}));
var Restrictions;
(function (Restrictions) {
    Restrictions.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Restrictions = exports.Restrictions || (exports.Restrictions = {}));
var ViewerCertificate;
(function (ViewerCertificate) {
    ViewerCertificate.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ViewerCertificate = exports.ViewerCertificate || (exports.ViewerCertificate = {}));
var DistributionConfig;
(function (DistributionConfig) {
    DistributionConfig.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Comment && { Comment: smithy_client_1.SENSITIVE_STRING }),
    });
})(DistributionConfig = exports.DistributionConfig || (exports.DistributionConfig = {}));
var CreateDistributionRequest;
(function (CreateDistributionRequest) {
    CreateDistributionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.DistributionConfig && {
            DistributionConfig: DistributionConfig.filterSensitiveLog(obj.DistributionConfig),
        }),
    });
})(CreateDistributionRequest = exports.CreateDistributionRequest || (exports.CreateDistributionRequest = {}));
var Distribution;
(function (Distribution) {
    Distribution.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.DistributionConfig && {
            DistributionConfig: DistributionConfig.filterSensitiveLog(obj.DistributionConfig),
        }),
    });
})(Distribution = exports.Distribution || (exports.Distribution = {}));
var CreateDistributionResult;
(function (CreateDistributionResult) {
    CreateDistributionResult.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Distribution && { Distribution: Distribution.filterSensitiveLog(obj.Distribution) }),
    });
})(CreateDistributionResult = exports.CreateDistributionResult || (exports.CreateDistributionResult = {}));
class DistributionAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "DistributionAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "DistributionAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, DistributionAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.DistributionAlreadyExists = DistributionAlreadyExists;
class IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior",
            $fault: "client",
            ...opts,
        });
        this.name = "IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior";
        this.$fault = "client";
        Object.setPrototypeOf(this, IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior.prototype);
        this.Message = opts.Message;
    }
}
exports.IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior = IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior;
class InvalidDefaultRootObject extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidDefaultRootObject",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidDefaultRootObject";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidDefaultRootObject.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidDefaultRootObject = InvalidDefaultRootObject;
class InvalidErrorCode extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidErrorCode",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidErrorCode";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidErrorCode.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidErrorCode = InvalidErrorCode;
class InvalidForwardCookies extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidForwardCookies",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidForwardCookies";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidForwardCookies.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidForwardCookies = InvalidForwardCookies;
class InvalidFunctionAssociation extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidFunctionAssociation",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidFunctionAssociation";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidFunctionAssociation.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidFunctionAssociation = InvalidFunctionAssociation;
class InvalidGeoRestrictionParameter extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidGeoRestrictionParameter",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidGeoRestrictionParameter";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidGeoRestrictionParameter.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidGeoRestrictionParameter = InvalidGeoRestrictionParameter;
class InvalidHeadersForS3Origin extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidHeadersForS3Origin",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidHeadersForS3Origin";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidHeadersForS3Origin.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidHeadersForS3Origin = InvalidHeadersForS3Origin;
class InvalidLambdaFunctionAssociation extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidLambdaFunctionAssociation",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidLambdaFunctionAssociation";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidLambdaFunctionAssociation.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidLambdaFunctionAssociation = InvalidLambdaFunctionAssociation;
class InvalidLocationCode extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidLocationCode",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidLocationCode";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidLocationCode.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidLocationCode = InvalidLocationCode;
class InvalidMinimumProtocolVersion extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidMinimumProtocolVersion",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidMinimumProtocolVersion";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidMinimumProtocolVersion.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidMinimumProtocolVersion = InvalidMinimumProtocolVersion;
class InvalidOrigin extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidOrigin",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidOrigin";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidOrigin.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidOrigin = InvalidOrigin;
class InvalidOriginAccessIdentity extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidOriginAccessIdentity",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidOriginAccessIdentity";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidOriginAccessIdentity.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidOriginAccessIdentity = InvalidOriginAccessIdentity;
class InvalidOriginKeepaliveTimeout extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidOriginKeepaliveTimeout",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidOriginKeepaliveTimeout";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidOriginKeepaliveTimeout.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidOriginKeepaliveTimeout = InvalidOriginKeepaliveTimeout;
class InvalidOriginReadTimeout extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidOriginReadTimeout",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidOriginReadTimeout";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidOriginReadTimeout.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidOriginReadTimeout = InvalidOriginReadTimeout;
class InvalidProtocolSettings extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidProtocolSettings",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidProtocolSettings";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidProtocolSettings.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidProtocolSettings = InvalidProtocolSettings;
class InvalidQueryStringParameters extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidQueryStringParameters",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidQueryStringParameters";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidQueryStringParameters.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidQueryStringParameters = InvalidQueryStringParameters;
class InvalidRelativePath extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidRelativePath",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidRelativePath";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidRelativePath.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidRelativePath = InvalidRelativePath;
class InvalidRequiredProtocol extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidRequiredProtocol",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidRequiredProtocol";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidRequiredProtocol.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidRequiredProtocol = InvalidRequiredProtocol;
class InvalidResponseCode extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidResponseCode",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidResponseCode";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidResponseCode.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidResponseCode = InvalidResponseCode;
class InvalidTTLOrder extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidTTLOrder",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidTTLOrder";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidTTLOrder.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidTTLOrder = InvalidTTLOrder;
class InvalidViewerCertificate extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidViewerCertificate",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidViewerCertificate";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidViewerCertificate.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidViewerCertificate = InvalidViewerCertificate;
class InvalidWebACLId extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidWebACLId",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidWebACLId";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidWebACLId.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidWebACLId = InvalidWebACLId;
class NoSuchCachePolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchCachePolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchCachePolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchCachePolicy = NoSuchCachePolicy;
class NoSuchFieldLevelEncryptionConfig extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchFieldLevelEncryptionConfig",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchFieldLevelEncryptionConfig";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchFieldLevelEncryptionConfig.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchFieldLevelEncryptionConfig = NoSuchFieldLevelEncryptionConfig;
class NoSuchOrigin extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchOrigin",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchOrigin";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchOrigin.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchOrigin = NoSuchOrigin;
class NoSuchOriginRequestPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchOriginRequestPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchOriginRequestPolicy = NoSuchOriginRequestPolicy;
class NoSuchRealtimeLogConfig extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchRealtimeLogConfig",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchRealtimeLogConfig";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchRealtimeLogConfig.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchRealtimeLogConfig = NoSuchRealtimeLogConfig;
class NoSuchResponseHeadersPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchResponseHeadersPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchResponseHeadersPolicy = NoSuchResponseHeadersPolicy;
class RealtimeLogConfigOwnerMismatch extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "RealtimeLogConfigOwnerMismatch",
            $fault: "client",
            ...opts,
        });
        this.name = "RealtimeLogConfigOwnerMismatch";
        this.$fault = "client";
        Object.setPrototypeOf(this, RealtimeLogConfigOwnerMismatch.prototype);
        this.Message = opts.Message;
    }
}
exports.RealtimeLogConfigOwnerMismatch = RealtimeLogConfigOwnerMismatch;
class TooManyCacheBehaviors extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyCacheBehaviors",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCacheBehaviors";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCacheBehaviors.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCacheBehaviors = TooManyCacheBehaviors;
class TooManyCertificates extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyCertificates",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCertificates";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCertificates.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCertificates = TooManyCertificates;
class TooManyCookieNamesInWhiteList extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyCookieNamesInWhiteList",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCookieNamesInWhiteList";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCookieNamesInWhiteList.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCookieNamesInWhiteList = TooManyCookieNamesInWhiteList;
class TooManyDistributions extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyDistributions",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributions";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributions.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributions = TooManyDistributions;
class TooManyDistributionsAssociatedToCachePolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToCachePolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsAssociatedToCachePolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsAssociatedToCachePolicy = TooManyDistributionsAssociatedToCachePolicy;
class TooManyDistributionsAssociatedToFieldLevelEncryptionConfig extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToFieldLevelEncryptionConfig",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsAssociatedToFieldLevelEncryptionConfig";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToFieldLevelEncryptionConfig.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsAssociatedToFieldLevelEncryptionConfig = TooManyDistributionsAssociatedToFieldLevelEncryptionConfig;
class TooManyDistributionsAssociatedToKeyGroup extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToKeyGroup",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsAssociatedToKeyGroup";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToKeyGroup.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsAssociatedToKeyGroup = TooManyDistributionsAssociatedToKeyGroup;
class TooManyDistributionsAssociatedToOriginRequestPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsAssociatedToOriginRequestPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsAssociatedToOriginRequestPolicy = TooManyDistributionsAssociatedToOriginRequestPolicy;
class TooManyDistributionsAssociatedToResponseHeadersPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsAssociatedToResponseHeadersPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsAssociatedToResponseHeadersPolicy = TooManyDistributionsAssociatedToResponseHeadersPolicy;
class TooManyDistributionsWithFunctionAssociations extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsWithFunctionAssociations",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsWithFunctionAssociations";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsWithFunctionAssociations.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsWithFunctionAssociations = TooManyDistributionsWithFunctionAssociations;
class TooManyDistributionsWithLambdaAssociations extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsWithLambdaAssociations",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsWithLambdaAssociations";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsWithLambdaAssociations.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsWithLambdaAssociations = TooManyDistributionsWithLambdaAssociations;
class TooManyDistributionsWithSingleFunctionARN extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsWithSingleFunctionARN",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsWithSingleFunctionARN";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsWithSingleFunctionARN.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyDistributionsWithSingleFunctionARN = TooManyDistributionsWithSingleFunctionARN;
class TooManyFunctionAssociations extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyFunctionAssociations",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFunctionAssociations";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFunctionAssociations.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFunctionAssociations = TooManyFunctionAssociations;
class TooManyHeadersInForwardedValues extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyHeadersInForwardedValues",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyHeadersInForwardedValues";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyHeadersInForwardedValues.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyHeadersInForwardedValues = TooManyHeadersInForwardedValues;
class TooManyKeyGroupsAssociatedToDistribution extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyKeyGroupsAssociatedToDistribution",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyKeyGroupsAssociatedToDistribution";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyKeyGroupsAssociatedToDistribution.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyKeyGroupsAssociatedToDistribution = TooManyKeyGroupsAssociatedToDistribution;
class TooManyLambdaFunctionAssociations extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyLambdaFunctionAssociations",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyLambdaFunctionAssociations";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyLambdaFunctionAssociations.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyLambdaFunctionAssociations = TooManyLambdaFunctionAssociations;
class TooManyOriginCustomHeaders extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyOriginCustomHeaders",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyOriginCustomHeaders";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyOriginCustomHeaders.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyOriginCustomHeaders = TooManyOriginCustomHeaders;
class TooManyOriginGroupsPerDistribution extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyOriginGroupsPerDistribution",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyOriginGroupsPerDistribution";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyOriginGroupsPerDistribution.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyOriginGroupsPerDistribution = TooManyOriginGroupsPerDistribution;
class TooManyOrigins extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyOrigins",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyOrigins";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyOrigins.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyOrigins = TooManyOrigins;
class TooManyQueryStringParameters extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyQueryStringParameters",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyQueryStringParameters";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyQueryStringParameters.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyQueryStringParameters = TooManyQueryStringParameters;
class TooManyTrustedSigners extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyTrustedSigners",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyTrustedSigners";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyTrustedSigners.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyTrustedSigners = TooManyTrustedSigners;
class TrustedKeyGroupDoesNotExist extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TrustedKeyGroupDoesNotExist",
            $fault: "client",
            ...opts,
        });
        this.name = "TrustedKeyGroupDoesNotExist";
        this.$fault = "client";
        Object.setPrototypeOf(this, TrustedKeyGroupDoesNotExist.prototype);
        this.Message = opts.Message;
    }
}
exports.TrustedKeyGroupDoesNotExist = TrustedKeyGroupDoesNotExist;
class TrustedSignerDoesNotExist extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TrustedSignerDoesNotExist",
            $fault: "client",
            ...opts,
        });
        this.name = "TrustedSignerDoesNotExist";
        this.$fault = "client";
        Object.setPrototypeOf(this, TrustedSignerDoesNotExist.prototype);
        this.Message = opts.Message;
    }
}
exports.TrustedSignerDoesNotExist = TrustedSignerDoesNotExist;
var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Tag = exports.Tag || (exports.Tag = {}));
var Tags;
(function (Tags) {
    Tags.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Tags = exports.Tags || (exports.Tags = {}));
var DistributionConfigWithTags;
(function (DistributionConfigWithTags) {
    DistributionConfigWithTags.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.DistributionConfig && {
            DistributionConfig: DistributionConfig.filterSensitiveLog(obj.DistributionConfig),
        }),
    });
})(DistributionConfigWithTags = exports.DistributionConfigWithTags || (exports.DistributionConfigWithTags = {}));
var CreateDistributionWithTagsRequest;
(function (CreateDistributionWithTagsRequest) {
    CreateDistributionWithTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.DistributionConfigWithTags && {
            DistributionConfigWithTags: DistributionConfigWithTags.filterSensitiveLog(obj.DistributionConfigWithTags),
        }),
    });
})(CreateDistributionWithTagsRequest = exports.CreateDistributionWithTagsRequest || (exports.CreateDistributionWithTagsRequest = {}));
var CreateDistributionWithTagsResult;
(function (CreateDistributionWithTagsResult) {
    CreateDistributionWithTagsResult.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Distribution && { Distribution: Distribution.filterSensitiveLog(obj.Distribution) }),
    });
})(CreateDistributionWithTagsResult = exports.CreateDistributionWithTagsResult || (exports.CreateDistributionWithTagsResult = {}));
class InvalidTagging extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidTagging",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidTagging";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidTagging.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidTagging = InvalidTagging;
var ContentTypeProfile;
(function (ContentTypeProfile) {
    ContentTypeProfile.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ContentTypeProfile = exports.ContentTypeProfile || (exports.ContentTypeProfile = {}));
var ContentTypeProfiles;
(function (ContentTypeProfiles) {
    ContentTypeProfiles.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ContentTypeProfiles = exports.ContentTypeProfiles || (exports.ContentTypeProfiles = {}));
var ContentTypeProfileConfig;
(function (ContentTypeProfileConfig) {
    ContentTypeProfileConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ContentTypeProfileConfig = exports.ContentTypeProfileConfig || (exports.ContentTypeProfileConfig = {}));
var QueryArgProfile;
(function (QueryArgProfile) {
    QueryArgProfile.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(QueryArgProfile = exports.QueryArgProfile || (exports.QueryArgProfile = {}));
var QueryArgProfiles;
(function (QueryArgProfiles) {
    QueryArgProfiles.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(QueryArgProfiles = exports.QueryArgProfiles || (exports.QueryArgProfiles = {}));
var QueryArgProfileConfig;
(function (QueryArgProfileConfig) {
    QueryArgProfileConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(QueryArgProfileConfig = exports.QueryArgProfileConfig || (exports.QueryArgProfileConfig = {}));
var FieldLevelEncryptionConfig;
(function (FieldLevelEncryptionConfig) {
    FieldLevelEncryptionConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FieldLevelEncryptionConfig = exports.FieldLevelEncryptionConfig || (exports.FieldLevelEncryptionConfig = {}));
var CreateFieldLevelEncryptionConfigRequest;
(function (CreateFieldLevelEncryptionConfigRequest) {
    CreateFieldLevelEncryptionConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateFieldLevelEncryptionConfigRequest = exports.CreateFieldLevelEncryptionConfigRequest || (exports.CreateFieldLevelEncryptionConfigRequest = {}));
var FieldLevelEncryption;
(function (FieldLevelEncryption) {
    FieldLevelEncryption.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FieldLevelEncryption = exports.FieldLevelEncryption || (exports.FieldLevelEncryption = {}));
var CreateFieldLevelEncryptionConfigResult;
(function (CreateFieldLevelEncryptionConfigResult) {
    CreateFieldLevelEncryptionConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateFieldLevelEncryptionConfigResult = exports.CreateFieldLevelEncryptionConfigResult || (exports.CreateFieldLevelEncryptionConfigResult = {}));
class FieldLevelEncryptionConfigAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionConfigAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "FieldLevelEncryptionConfigAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, FieldLevelEncryptionConfigAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.FieldLevelEncryptionConfigAlreadyExists = FieldLevelEncryptionConfigAlreadyExists;
class NoSuchFieldLevelEncryptionProfile extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchFieldLevelEncryptionProfile",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchFieldLevelEncryptionProfile";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchFieldLevelEncryptionProfile.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchFieldLevelEncryptionProfile = NoSuchFieldLevelEncryptionProfile;
class QueryArgProfileEmpty extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "QueryArgProfileEmpty",
            $fault: "client",
            ...opts,
        });
        this.name = "QueryArgProfileEmpty";
        this.$fault = "client";
        Object.setPrototypeOf(this, QueryArgProfileEmpty.prototype);
        this.Message = opts.Message;
    }
}
exports.QueryArgProfileEmpty = QueryArgProfileEmpty;
class TooManyFieldLevelEncryptionConfigs extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionConfigs",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionConfigs";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionConfigs.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionConfigs = TooManyFieldLevelEncryptionConfigs;
class TooManyFieldLevelEncryptionContentTypeProfiles extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionContentTypeProfiles",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionContentTypeProfiles";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionContentTypeProfiles.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionContentTypeProfiles = TooManyFieldLevelEncryptionContentTypeProfiles;
class TooManyFieldLevelEncryptionQueryArgProfiles extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionQueryArgProfiles",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionQueryArgProfiles";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionQueryArgProfiles.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionQueryArgProfiles = TooManyFieldLevelEncryptionQueryArgProfiles;
var FieldPatterns;
(function (FieldPatterns) {
    FieldPatterns.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FieldPatterns = exports.FieldPatterns || (exports.FieldPatterns = {}));
var EncryptionEntity;
(function (EncryptionEntity) {
    EncryptionEntity.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EncryptionEntity = exports.EncryptionEntity || (exports.EncryptionEntity = {}));
var EncryptionEntities;
(function (EncryptionEntities) {
    EncryptionEntities.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EncryptionEntities = exports.EncryptionEntities || (exports.EncryptionEntities = {}));
var FieldLevelEncryptionProfileConfig;
(function (FieldLevelEncryptionProfileConfig) {
    FieldLevelEncryptionProfileConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FieldLevelEncryptionProfileConfig = exports.FieldLevelEncryptionProfileConfig || (exports.FieldLevelEncryptionProfileConfig = {}));
var CreateFieldLevelEncryptionProfileRequest;
(function (CreateFieldLevelEncryptionProfileRequest) {
    CreateFieldLevelEncryptionProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateFieldLevelEncryptionProfileRequest = exports.CreateFieldLevelEncryptionProfileRequest || (exports.CreateFieldLevelEncryptionProfileRequest = {}));
var FieldLevelEncryptionProfile;
(function (FieldLevelEncryptionProfile) {
    FieldLevelEncryptionProfile.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FieldLevelEncryptionProfile = exports.FieldLevelEncryptionProfile || (exports.FieldLevelEncryptionProfile = {}));
var CreateFieldLevelEncryptionProfileResult;
(function (CreateFieldLevelEncryptionProfileResult) {
    CreateFieldLevelEncryptionProfileResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateFieldLevelEncryptionProfileResult = exports.CreateFieldLevelEncryptionProfileResult || (exports.CreateFieldLevelEncryptionProfileResult = {}));
class FieldLevelEncryptionProfileAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionProfileAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "FieldLevelEncryptionProfileAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, FieldLevelEncryptionProfileAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.FieldLevelEncryptionProfileAlreadyExists = FieldLevelEncryptionProfileAlreadyExists;
class FieldLevelEncryptionProfileSizeExceeded extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionProfileSizeExceeded",
            $fault: "client",
            ...opts,
        });
        this.name = "FieldLevelEncryptionProfileSizeExceeded";
        this.$fault = "client";
        Object.setPrototypeOf(this, FieldLevelEncryptionProfileSizeExceeded.prototype);
        this.Message = opts.Message;
    }
}
exports.FieldLevelEncryptionProfileSizeExceeded = FieldLevelEncryptionProfileSizeExceeded;
class NoSuchPublicKey extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchPublicKey",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchPublicKey";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchPublicKey.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchPublicKey = NoSuchPublicKey;
class TooManyFieldLevelEncryptionEncryptionEntities extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionEncryptionEntities",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionEncryptionEntities";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionEncryptionEntities.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionEncryptionEntities = TooManyFieldLevelEncryptionEncryptionEntities;
class TooManyFieldLevelEncryptionFieldPatterns extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionFieldPatterns",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionFieldPatterns";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionFieldPatterns.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionFieldPatterns = TooManyFieldLevelEncryptionFieldPatterns;
class TooManyFieldLevelEncryptionProfiles extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionProfiles",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionProfiles";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionProfiles.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFieldLevelEncryptionProfiles = TooManyFieldLevelEncryptionProfiles;
var FunctionRuntime;
(function (FunctionRuntime) {
    FunctionRuntime["cloudfront_js_1_0"] = "cloudfront-js-1.0";
})(FunctionRuntime = exports.FunctionRuntime || (exports.FunctionRuntime = {}));
var FunctionConfig;
(function (FunctionConfig) {
    FunctionConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FunctionConfig = exports.FunctionConfig || (exports.FunctionConfig = {}));
var CreateFunctionRequest;
(function (CreateFunctionRequest) {
    CreateFunctionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.FunctionCode && { FunctionCode: smithy_client_1.SENSITIVE_STRING }),
    });
})(CreateFunctionRequest = exports.CreateFunctionRequest || (exports.CreateFunctionRequest = {}));
var FunctionStage;
(function (FunctionStage) {
    FunctionStage["DEVELOPMENT"] = "DEVELOPMENT";
    FunctionStage["LIVE"] = "LIVE";
})(FunctionStage = exports.FunctionStage || (exports.FunctionStage = {}));
var FunctionMetadata;
(function (FunctionMetadata) {
    FunctionMetadata.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FunctionMetadata = exports.FunctionMetadata || (exports.FunctionMetadata = {}));
var FunctionSummary;
(function (FunctionSummary) {
    FunctionSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FunctionSummary = exports.FunctionSummary || (exports.FunctionSummary = {}));
var CreateFunctionResult;
(function (CreateFunctionResult) {
    CreateFunctionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateFunctionResult = exports.CreateFunctionResult || (exports.CreateFunctionResult = {}));
class FunctionAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "FunctionAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "FunctionAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, FunctionAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.FunctionAlreadyExists = FunctionAlreadyExists;
class FunctionSizeLimitExceeded extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "FunctionSizeLimitExceeded",
            $fault: "client",
            ...opts,
        });
        this.name = "FunctionSizeLimitExceeded";
        this.$fault = "client";
        Object.setPrototypeOf(this, FunctionSizeLimitExceeded.prototype);
        this.Message = opts.Message;
    }
}
exports.FunctionSizeLimitExceeded = FunctionSizeLimitExceeded;
class TooManyFunctions extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyFunctions",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFunctions";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFunctions.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyFunctions = TooManyFunctions;
class UnsupportedOperation extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "UnsupportedOperation",
            $fault: "client",
            ...opts,
        });
        this.name = "UnsupportedOperation";
        this.$fault = "client";
        Object.setPrototypeOf(this, UnsupportedOperation.prototype);
        this.Message = opts.Message;
    }
}
exports.UnsupportedOperation = UnsupportedOperation;
var Paths;
(function (Paths) {
    Paths.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Paths = exports.Paths || (exports.Paths = {}));
var InvalidationBatch;
(function (InvalidationBatch) {
    InvalidationBatch.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidationBatch = exports.InvalidationBatch || (exports.InvalidationBatch = {}));
var CreateInvalidationRequest;
(function (CreateInvalidationRequest) {
    CreateInvalidationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateInvalidationRequest = exports.CreateInvalidationRequest || (exports.CreateInvalidationRequest = {}));
var Invalidation;
(function (Invalidation) {
    Invalidation.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Invalidation = exports.Invalidation || (exports.Invalidation = {}));
var CreateInvalidationResult;
(function (CreateInvalidationResult) {
    CreateInvalidationResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateInvalidationResult = exports.CreateInvalidationResult || (exports.CreateInvalidationResult = {}));
class TooManyInvalidationsInProgress extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyInvalidationsInProgress",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyInvalidationsInProgress";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyInvalidationsInProgress.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyInvalidationsInProgress = TooManyInvalidationsInProgress;
var KeyGroupConfig;
(function (KeyGroupConfig) {
    KeyGroupConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(KeyGroupConfig = exports.KeyGroupConfig || (exports.KeyGroupConfig = {}));
var CreateKeyGroupRequest;
(function (CreateKeyGroupRequest) {
    CreateKeyGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateKeyGroupRequest = exports.CreateKeyGroupRequest || (exports.CreateKeyGroupRequest = {}));
var KeyGroup;
(function (KeyGroup) {
    KeyGroup.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(KeyGroup = exports.KeyGroup || (exports.KeyGroup = {}));
var CreateKeyGroupResult;
(function (CreateKeyGroupResult) {
    CreateKeyGroupResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateKeyGroupResult = exports.CreateKeyGroupResult || (exports.CreateKeyGroupResult = {}));
class KeyGroupAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "KeyGroupAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "KeyGroupAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, KeyGroupAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.KeyGroupAlreadyExists = KeyGroupAlreadyExists;
class TooManyKeyGroups extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyKeyGroups",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyKeyGroups";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyKeyGroups.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyKeyGroups = TooManyKeyGroups;
class TooManyPublicKeysInKeyGroup extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyPublicKeysInKeyGroup",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyPublicKeysInKeyGroup";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyPublicKeysInKeyGroup.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyPublicKeysInKeyGroup = TooManyPublicKeysInKeyGroup;
var RealtimeMetricsSubscriptionStatus;
(function (RealtimeMetricsSubscriptionStatus) {
    RealtimeMetricsSubscriptionStatus["Disabled"] = "Disabled";
    RealtimeMetricsSubscriptionStatus["Enabled"] = "Enabled";
})(RealtimeMetricsSubscriptionStatus = exports.RealtimeMetricsSubscriptionStatus || (exports.RealtimeMetricsSubscriptionStatus = {}));
var RealtimeMetricsSubscriptionConfig;
(function (RealtimeMetricsSubscriptionConfig) {
    RealtimeMetricsSubscriptionConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RealtimeMetricsSubscriptionConfig = exports.RealtimeMetricsSubscriptionConfig || (exports.RealtimeMetricsSubscriptionConfig = {}));
var MonitoringSubscription;
(function (MonitoringSubscription) {
    MonitoringSubscription.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(MonitoringSubscription = exports.MonitoringSubscription || (exports.MonitoringSubscription = {}));
var CreateMonitoringSubscriptionRequest;
(function (CreateMonitoringSubscriptionRequest) {
    CreateMonitoringSubscriptionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateMonitoringSubscriptionRequest = exports.CreateMonitoringSubscriptionRequest || (exports.CreateMonitoringSubscriptionRequest = {}));
var CreateMonitoringSubscriptionResult;
(function (CreateMonitoringSubscriptionResult) {
    CreateMonitoringSubscriptionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateMonitoringSubscriptionResult = exports.CreateMonitoringSubscriptionResult || (exports.CreateMonitoringSubscriptionResult = {}));
var OriginRequestPolicyCookiesConfig;
(function (OriginRequestPolicyCookiesConfig) {
    OriginRequestPolicyCookiesConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginRequestPolicyCookiesConfig = exports.OriginRequestPolicyCookiesConfig || (exports.OriginRequestPolicyCookiesConfig = {}));
var OriginRequestPolicyHeadersConfig;
(function (OriginRequestPolicyHeadersConfig) {
    OriginRequestPolicyHeadersConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginRequestPolicyHeadersConfig = exports.OriginRequestPolicyHeadersConfig || (exports.OriginRequestPolicyHeadersConfig = {}));
var OriginRequestPolicyQueryStringsConfig;
(function (OriginRequestPolicyQueryStringsConfig) {
    OriginRequestPolicyQueryStringsConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginRequestPolicyQueryStringsConfig = exports.OriginRequestPolicyQueryStringsConfig || (exports.OriginRequestPolicyQueryStringsConfig = {}));
var OriginRequestPolicyConfig;
(function (OriginRequestPolicyConfig) {
    OriginRequestPolicyConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginRequestPolicyConfig = exports.OriginRequestPolicyConfig || (exports.OriginRequestPolicyConfig = {}));
var CreateOriginRequestPolicyRequest;
(function (CreateOriginRequestPolicyRequest) {
    CreateOriginRequestPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateOriginRequestPolicyRequest = exports.CreateOriginRequestPolicyRequest || (exports.CreateOriginRequestPolicyRequest = {}));
var OriginRequestPolicy;
(function (OriginRequestPolicy) {
    OriginRequestPolicy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OriginRequestPolicy = exports.OriginRequestPolicy || (exports.OriginRequestPolicy = {}));
var CreateOriginRequestPolicyResult;
(function (CreateOriginRequestPolicyResult) {
    CreateOriginRequestPolicyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateOriginRequestPolicyResult = exports.CreateOriginRequestPolicyResult || (exports.CreateOriginRequestPolicyResult = {}));
class OriginRequestPolicyAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "OriginRequestPolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "OriginRequestPolicyAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, OriginRequestPolicyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.OriginRequestPolicyAlreadyExists = OriginRequestPolicyAlreadyExists;
class TooManyCookiesInOriginRequestPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyCookiesInOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCookiesInOriginRequestPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCookiesInOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCookiesInOriginRequestPolicy = TooManyCookiesInOriginRequestPolicy;
class TooManyHeadersInOriginRequestPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyHeadersInOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyHeadersInOriginRequestPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyHeadersInOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyHeadersInOriginRequestPolicy = TooManyHeadersInOriginRequestPolicy;
class TooManyOriginRequestPolicies extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyOriginRequestPolicies",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyOriginRequestPolicies";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyOriginRequestPolicies.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyOriginRequestPolicies = TooManyOriginRequestPolicies;
class TooManyQueryStringsInOriginRequestPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyQueryStringsInOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyQueryStringsInOriginRequestPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyQueryStringsInOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyQueryStringsInOriginRequestPolicy = TooManyQueryStringsInOriginRequestPolicy;
var PublicKeyConfig;
(function (PublicKeyConfig) {
    PublicKeyConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PublicKeyConfig = exports.PublicKeyConfig || (exports.PublicKeyConfig = {}));
var CreatePublicKeyRequest;
(function (CreatePublicKeyRequest) {
    CreatePublicKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreatePublicKeyRequest = exports.CreatePublicKeyRequest || (exports.CreatePublicKeyRequest = {}));
var PublicKey;
(function (PublicKey) {
    PublicKey.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PublicKey = exports.PublicKey || (exports.PublicKey = {}));
var CreatePublicKeyResult;
(function (CreatePublicKeyResult) {
    CreatePublicKeyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreatePublicKeyResult = exports.CreatePublicKeyResult || (exports.CreatePublicKeyResult = {}));
class PublicKeyAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "PublicKeyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "PublicKeyAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, PublicKeyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.PublicKeyAlreadyExists = PublicKeyAlreadyExists;
class TooManyPublicKeys extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyPublicKeys",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyPublicKeys";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyPublicKeys.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyPublicKeys = TooManyPublicKeys;
var KinesisStreamConfig;
(function (KinesisStreamConfig) {
    KinesisStreamConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(KinesisStreamConfig = exports.KinesisStreamConfig || (exports.KinesisStreamConfig = {}));
var EndPoint;
(function (EndPoint) {
    EndPoint.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EndPoint = exports.EndPoint || (exports.EndPoint = {}));
var CreateRealtimeLogConfigRequest;
(function (CreateRealtimeLogConfigRequest) {
    CreateRealtimeLogConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateRealtimeLogConfigRequest = exports.CreateRealtimeLogConfigRequest || (exports.CreateRealtimeLogConfigRequest = {}));
var RealtimeLogConfig;
(function (RealtimeLogConfig) {
    RealtimeLogConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RealtimeLogConfig = exports.RealtimeLogConfig || (exports.RealtimeLogConfig = {}));
var CreateRealtimeLogConfigResult;
(function (CreateRealtimeLogConfigResult) {
    CreateRealtimeLogConfigResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateRealtimeLogConfigResult = exports.CreateRealtimeLogConfigResult || (exports.CreateRealtimeLogConfigResult = {}));
class RealtimeLogConfigAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "RealtimeLogConfigAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "RealtimeLogConfigAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, RealtimeLogConfigAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.RealtimeLogConfigAlreadyExists = RealtimeLogConfigAlreadyExists;
class TooManyRealtimeLogConfigs extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyRealtimeLogConfigs",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyRealtimeLogConfigs";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyRealtimeLogConfigs.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyRealtimeLogConfigs = TooManyRealtimeLogConfigs;
var ResponseHeadersPolicyAccessControlAllowHeaders;
(function (ResponseHeadersPolicyAccessControlAllowHeaders) {
    ResponseHeadersPolicyAccessControlAllowHeaders.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyAccessControlAllowHeaders = exports.ResponseHeadersPolicyAccessControlAllowHeaders || (exports.ResponseHeadersPolicyAccessControlAllowHeaders = {}));
var ResponseHeadersPolicyAccessControlAllowMethods;
(function (ResponseHeadersPolicyAccessControlAllowMethods) {
    ResponseHeadersPolicyAccessControlAllowMethods.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyAccessControlAllowMethods = exports.ResponseHeadersPolicyAccessControlAllowMethods || (exports.ResponseHeadersPolicyAccessControlAllowMethods = {}));
var ResponseHeadersPolicyAccessControlAllowOrigins;
(function (ResponseHeadersPolicyAccessControlAllowOrigins) {
    ResponseHeadersPolicyAccessControlAllowOrigins.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyAccessControlAllowOrigins = exports.ResponseHeadersPolicyAccessControlAllowOrigins || (exports.ResponseHeadersPolicyAccessControlAllowOrigins = {}));
var ResponseHeadersPolicyAccessControlExposeHeaders;
(function (ResponseHeadersPolicyAccessControlExposeHeaders) {
    ResponseHeadersPolicyAccessControlExposeHeaders.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyAccessControlExposeHeaders = exports.ResponseHeadersPolicyAccessControlExposeHeaders || (exports.ResponseHeadersPolicyAccessControlExposeHeaders = {}));
var ResponseHeadersPolicyCorsConfig;
(function (ResponseHeadersPolicyCorsConfig) {
    ResponseHeadersPolicyCorsConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyCorsConfig = exports.ResponseHeadersPolicyCorsConfig || (exports.ResponseHeadersPolicyCorsConfig = {}));
var ResponseHeadersPolicyCustomHeader;
(function (ResponseHeadersPolicyCustomHeader) {
    ResponseHeadersPolicyCustomHeader.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyCustomHeader = exports.ResponseHeadersPolicyCustomHeader || (exports.ResponseHeadersPolicyCustomHeader = {}));
var ResponseHeadersPolicyCustomHeadersConfig;
(function (ResponseHeadersPolicyCustomHeadersConfig) {
    ResponseHeadersPolicyCustomHeadersConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyCustomHeadersConfig = exports.ResponseHeadersPolicyCustomHeadersConfig || (exports.ResponseHeadersPolicyCustomHeadersConfig = {}));
var ResponseHeadersPolicyContentSecurityPolicy;
(function (ResponseHeadersPolicyContentSecurityPolicy) {
    ResponseHeadersPolicyContentSecurityPolicy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyContentSecurityPolicy = exports.ResponseHeadersPolicyContentSecurityPolicy || (exports.ResponseHeadersPolicyContentSecurityPolicy = {}));
var ResponseHeadersPolicyContentTypeOptions;
(function (ResponseHeadersPolicyContentTypeOptions) {
    ResponseHeadersPolicyContentTypeOptions.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyContentTypeOptions = exports.ResponseHeadersPolicyContentTypeOptions || (exports.ResponseHeadersPolicyContentTypeOptions = {}));
var FrameOptionsList;
(function (FrameOptionsList) {
    FrameOptionsList["DENY"] = "DENY";
    FrameOptionsList["SAMEORIGIN"] = "SAMEORIGIN";
})(FrameOptionsList = exports.FrameOptionsList || (exports.FrameOptionsList = {}));
var ResponseHeadersPolicyFrameOptions;
(function (ResponseHeadersPolicyFrameOptions) {
    ResponseHeadersPolicyFrameOptions.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyFrameOptions = exports.ResponseHeadersPolicyFrameOptions || (exports.ResponseHeadersPolicyFrameOptions = {}));
var ReferrerPolicyList;
(function (ReferrerPolicyList) {
    ReferrerPolicyList["no_referrer"] = "no-referrer";
    ReferrerPolicyList["no_referrer_when_downgrade"] = "no-referrer-when-downgrade";
    ReferrerPolicyList["origin"] = "origin";
    ReferrerPolicyList["origin_when_cross_origin"] = "origin-when-cross-origin";
    ReferrerPolicyList["same_origin"] = "same-origin";
    ReferrerPolicyList["strict_origin"] = "strict-origin";
    ReferrerPolicyList["strict_origin_when_cross_origin"] = "strict-origin-when-cross-origin";
    ReferrerPolicyList["unsafe_url"] = "unsafe-url";
})(ReferrerPolicyList = exports.ReferrerPolicyList || (exports.ReferrerPolicyList = {}));
var ResponseHeadersPolicyReferrerPolicy;
(function (ResponseHeadersPolicyReferrerPolicy) {
    ResponseHeadersPolicyReferrerPolicy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyReferrerPolicy = exports.ResponseHeadersPolicyReferrerPolicy || (exports.ResponseHeadersPolicyReferrerPolicy = {}));
var ResponseHeadersPolicyStrictTransportSecurity;
(function (ResponseHeadersPolicyStrictTransportSecurity) {
    ResponseHeadersPolicyStrictTransportSecurity.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyStrictTransportSecurity = exports.ResponseHeadersPolicyStrictTransportSecurity || (exports.ResponseHeadersPolicyStrictTransportSecurity = {}));
var ResponseHeadersPolicyXSSProtection;
(function (ResponseHeadersPolicyXSSProtection) {
    ResponseHeadersPolicyXSSProtection.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyXSSProtection = exports.ResponseHeadersPolicyXSSProtection || (exports.ResponseHeadersPolicyXSSProtection = {}));
var ResponseHeadersPolicySecurityHeadersConfig;
(function (ResponseHeadersPolicySecurityHeadersConfig) {
    ResponseHeadersPolicySecurityHeadersConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicySecurityHeadersConfig = exports.ResponseHeadersPolicySecurityHeadersConfig || (exports.ResponseHeadersPolicySecurityHeadersConfig = {}));
var ResponseHeadersPolicyServerTimingHeadersConfig;
(function (ResponseHeadersPolicyServerTimingHeadersConfig) {
    ResponseHeadersPolicyServerTimingHeadersConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyServerTimingHeadersConfig = exports.ResponseHeadersPolicyServerTimingHeadersConfig || (exports.ResponseHeadersPolicyServerTimingHeadersConfig = {}));
var ResponseHeadersPolicyConfig;
(function (ResponseHeadersPolicyConfig) {
    ResponseHeadersPolicyConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicyConfig = exports.ResponseHeadersPolicyConfig || (exports.ResponseHeadersPolicyConfig = {}));
var CreateResponseHeadersPolicyRequest;
(function (CreateResponseHeadersPolicyRequest) {
    CreateResponseHeadersPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateResponseHeadersPolicyRequest = exports.CreateResponseHeadersPolicyRequest || (exports.CreateResponseHeadersPolicyRequest = {}));
var ResponseHeadersPolicy;
(function (ResponseHeadersPolicy) {
    ResponseHeadersPolicy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResponseHeadersPolicy = exports.ResponseHeadersPolicy || (exports.ResponseHeadersPolicy = {}));
var CreateResponseHeadersPolicyResult;
(function (CreateResponseHeadersPolicyResult) {
    CreateResponseHeadersPolicyResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateResponseHeadersPolicyResult = exports.CreateResponseHeadersPolicyResult || (exports.CreateResponseHeadersPolicyResult = {}));
class ResponseHeadersPolicyAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "ResponseHeadersPolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "ResponseHeadersPolicyAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResponseHeadersPolicyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.ResponseHeadersPolicyAlreadyExists = ResponseHeadersPolicyAlreadyExists;
class TooLongCSPInResponseHeadersPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooLongCSPInResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooLongCSPInResponseHeadersPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooLongCSPInResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooLongCSPInResponseHeadersPolicy = TooLongCSPInResponseHeadersPolicy;
class TooManyCustomHeadersInResponseHeadersPolicy extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyCustomHeadersInResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCustomHeadersInResponseHeadersPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCustomHeadersInResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyCustomHeadersInResponseHeadersPolicy = TooManyCustomHeadersInResponseHeadersPolicy;
class TooManyResponseHeadersPolicies extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyResponseHeadersPolicies",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyResponseHeadersPolicies";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyResponseHeadersPolicies.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyResponseHeadersPolicies = TooManyResponseHeadersPolicies;
var StreamingLoggingConfig;
(function (StreamingLoggingConfig) {
    StreamingLoggingConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StreamingLoggingConfig = exports.StreamingLoggingConfig || (exports.StreamingLoggingConfig = {}));
var S3Origin;
(function (S3Origin) {
    S3Origin.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(S3Origin = exports.S3Origin || (exports.S3Origin = {}));
var StreamingDistributionConfig;
(function (StreamingDistributionConfig) {
    StreamingDistributionConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StreamingDistributionConfig = exports.StreamingDistributionConfig || (exports.StreamingDistributionConfig = {}));
var CreateStreamingDistributionRequest;
(function (CreateStreamingDistributionRequest) {
    CreateStreamingDistributionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateStreamingDistributionRequest = exports.CreateStreamingDistributionRequest || (exports.CreateStreamingDistributionRequest = {}));
var StreamingDistribution;
(function (StreamingDistribution) {
    StreamingDistribution.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StreamingDistribution = exports.StreamingDistribution || (exports.StreamingDistribution = {}));
var CreateStreamingDistributionResult;
(function (CreateStreamingDistributionResult) {
    CreateStreamingDistributionResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateStreamingDistributionResult = exports.CreateStreamingDistributionResult || (exports.CreateStreamingDistributionResult = {}));
class StreamingDistributionAlreadyExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "StreamingDistributionAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "StreamingDistributionAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, StreamingDistributionAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
exports.StreamingDistributionAlreadyExists = StreamingDistributionAlreadyExists;
class TooManyStreamingDistributionCNAMEs extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyStreamingDistributionCNAMEs",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyStreamingDistributionCNAMEs";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyStreamingDistributionCNAMEs.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyStreamingDistributionCNAMEs = TooManyStreamingDistributionCNAMEs;
class TooManyStreamingDistributions extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "TooManyStreamingDistributions",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyStreamingDistributions";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyStreamingDistributions.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyStreamingDistributions = TooManyStreamingDistributions;
var StreamingDistributionConfigWithTags;
(function (StreamingDistributionConfigWithTags) {
    StreamingDistributionConfigWithTags.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StreamingDistributionConfigWithTags = exports.StreamingDistributionConfigWithTags || (exports.StreamingDistributionConfigWithTags = {}));
var CreateStreamingDistributionWithTagsRequest;
(function (CreateStreamingDistributionWithTagsRequest) {
    CreateStreamingDistributionWithTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateStreamingDistributionWithTagsRequest = exports.CreateStreamingDistributionWithTagsRequest || (exports.CreateStreamingDistributionWithTagsRequest = {}));
var CreateStreamingDistributionWithTagsResult;
(function (CreateStreamingDistributionWithTagsResult) {
    CreateStreamingDistributionWithTagsResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateStreamingDistributionWithTagsResult = exports.CreateStreamingDistributionWithTagsResult || (exports.CreateStreamingDistributionWithTagsResult = {}));
var DeleteCachePolicyRequest;
(function (DeleteCachePolicyRequest) {
    DeleteCachePolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteCachePolicyRequest = exports.DeleteCachePolicyRequest || (exports.DeleteCachePolicyRequest = {}));
class IllegalDelete extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "IllegalDelete",
            $fault: "client",
            ...opts,
        });
        this.name = "IllegalDelete";
        this.$fault = "client";
        Object.setPrototypeOf(this, IllegalDelete.prototype);
        this.Message = opts.Message;
    }
}
exports.IllegalDelete = IllegalDelete;
class InvalidIfMatchVersion extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "InvalidIfMatchVersion",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidIfMatchVersion";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidIfMatchVersion.prototype);
        this.Message = opts.Message;
    }
}
exports.InvalidIfMatchVersion = InvalidIfMatchVersion;
class PreconditionFailed extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "PreconditionFailed",
            $fault: "client",
            ...opts,
        });
        this.name = "PreconditionFailed";
        this.$fault = "client";
        Object.setPrototypeOf(this, PreconditionFailed.prototype);
        this.Message = opts.Message;
    }
}
exports.PreconditionFailed = PreconditionFailed;
class CloudFrontOriginAccessIdentityInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "CloudFrontOriginAccessIdentityInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "CloudFrontOriginAccessIdentityInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, CloudFrontOriginAccessIdentityInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.CloudFrontOriginAccessIdentityInUse = CloudFrontOriginAccessIdentityInUse;
var DeleteCloudFrontOriginAccessIdentityRequest;
(function (DeleteCloudFrontOriginAccessIdentityRequest) {
    DeleteCloudFrontOriginAccessIdentityRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteCloudFrontOriginAccessIdentityRequest = exports.DeleteCloudFrontOriginAccessIdentityRequest || (exports.DeleteCloudFrontOriginAccessIdentityRequest = {}));
class NoSuchCloudFrontOriginAccessIdentity extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchCloudFrontOriginAccessIdentity",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchCloudFrontOriginAccessIdentity";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchCloudFrontOriginAccessIdentity.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchCloudFrontOriginAccessIdentity = NoSuchCloudFrontOriginAccessIdentity;
var DeleteDistributionRequest;
(function (DeleteDistributionRequest) {
    DeleteDistributionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteDistributionRequest = exports.DeleteDistributionRequest || (exports.DeleteDistributionRequest = {}));
class DistributionNotDisabled extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "DistributionNotDisabled",
            $fault: "client",
            ...opts,
        });
        this.name = "DistributionNotDisabled";
        this.$fault = "client";
        Object.setPrototypeOf(this, DistributionNotDisabled.prototype);
        this.Message = opts.Message;
    }
}
exports.DistributionNotDisabled = DistributionNotDisabled;
var DeleteFieldLevelEncryptionConfigRequest;
(function (DeleteFieldLevelEncryptionConfigRequest) {
    DeleteFieldLevelEncryptionConfigRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteFieldLevelEncryptionConfigRequest = exports.DeleteFieldLevelEncryptionConfigRequest || (exports.DeleteFieldLevelEncryptionConfigRequest = {}));
class FieldLevelEncryptionConfigInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionConfigInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "FieldLevelEncryptionConfigInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, FieldLevelEncryptionConfigInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.FieldLevelEncryptionConfigInUse = FieldLevelEncryptionConfigInUse;
var DeleteFieldLevelEncryptionProfileRequest;
(function (DeleteFieldLevelEncryptionProfileRequest) {
    DeleteFieldLevelEncryptionProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteFieldLevelEncryptionProfileRequest = exports.DeleteFieldLevelEncryptionProfileRequest || (exports.DeleteFieldLevelEncryptionProfileRequest = {}));
class FieldLevelEncryptionProfileInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionProfileInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "FieldLevelEncryptionProfileInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, FieldLevelEncryptionProfileInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.FieldLevelEncryptionProfileInUse = FieldLevelEncryptionProfileInUse;
var DeleteFunctionRequest;
(function (DeleteFunctionRequest) {
    DeleteFunctionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteFunctionRequest = exports.DeleteFunctionRequest || (exports.DeleteFunctionRequest = {}));
class FunctionInUse extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "FunctionInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "FunctionInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, FunctionInUse.prototype);
        this.Message = opts.Message;
    }
}
exports.FunctionInUse = FunctionInUse;
class NoSuchFunctionExists extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchFunctionExists",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchFunctionExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchFunctionExists.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchFunctionExists = NoSuchFunctionExists;
var DeleteKeyGroupRequest;
(function (DeleteKeyGroupRequest) {
    DeleteKeyGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteKeyGroupRequest = exports.DeleteKeyGroupRequest || (exports.DeleteKeyGroupRequest = {}));
class NoSuchResource extends CloudFrontServiceException_1.CloudFrontServiceException {
    constructor(opts) {
        super({
            name: "NoSuchResource",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchResource";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchResource.prototype);
        this.Message = opts.Message;
    }
}
exports.NoSuchResource = NoSuchResource;
