// roamjs-components/backend/emailError.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const sendEmail_1 = tslib_1.__importDefault(require("aws-sdk-plus/dist/sendEmail"));
const react_1 = tslib_1.__importDefault(require("react"));
const emailError = (subject, e, moreBody) => {
    var _a, _b;
    const asAxios = e;
    return (0, sendEmail_1.default)({
        to: process.env.ROAMJS_EMAIL,
        from: "support@roamjs.com",
        subject: `RoamJS Error: ${subject}`,
        body: react_1.default.createElement("div", {
            style: {
                margin: "0 auto",
                maxWidth: 600,
                fontFamily: `"Proxima Nova","proxima-nova",Helvetica,Arial sans-serif`,
                padding: `20px 0`,
            },
        }, react_1.default.createElement("div", {
            style: {
                width: "80%",
                margin: "0 auto",
                paddingBottom: 20,
                borderBottom: "1px dashed #dadada",
                textAlign: "center",
            },
        }, react_1.default.createElement("img", {
            src: "https://roamjs.com/images/logo-low-res.png",
            width: 128,
        })), react_1.default.createElement("div", {
            style: {
                width: "80%",
                margin: "30px auto",
                fontSize: 16,
            },
        }, react_1.default.createElement("h3", {}, `An error was thrown in a RoamJS Lambda`), react_1.default.createElement("p", {}, `${e.name}: ${typeof ((_a = asAxios.response) === null || _a === void 0 ? void 0 : _a.data) === "object"
            ? asAxios.response.data.message ||
                JSON.stringify(asAxios.response.data)
            : ((_b = asAxios.response) === null || _b === void 0 ? void 0 : _b.data) || e.message}`), react_1.default.createElement("p", {}, e.stack), ...(moreBody ? [react_1.default.createElement("div", {}, moreBody)] : [])), react_1.default.createElement("div", {
            style: {
                width: "80%",
                margin: "30px auto",
                borderTop: "1px dashed #dadada",
                display: "flex",
                color: "#a8a8a8",
                paddingTop: 15,
            },
        }, react_1.default.createElement("div", { style: { width: "50%" } }, "Sent From ", react_1.default.createElement("a", {
            href: "https://roamjs.com",
            style: { color: "#3ba4dc", textDecoration: "none" },
        }, "RoamJS")), react_1.default.createElement("div", { style: { width: "50%", textAlign: "right" } }, react_1.default.createElement("a", {
            href: "mailto:support@roamjs.com",
            style: { color: "#3ba4dc", textDecoration: "none" },
        }, "Contact Support")))),
    });
};
exports.default = emailError;
