// @aws-sdk/client-cloudfront/dist-es/commands/CreateMonitoringSubscriptionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateMonitoringSubscriptionRequest, CreateMonitoringSubscriptionResult } from "../models/models_0";
import { deserializeAws_restXmlCreateMonitoringSubscriptionCommand, serializeAws_restXmlCreateMonitoringSubscriptionCommand, } from "../protocols/Aws_restXml";
var CreateMonitoringSubscriptionCommand = (function (_super) {
    __extends(CreateMonitoringSubscriptionCommand, _super);
    function CreateMonitoringSubscriptionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateMonitoringSubscriptionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "CreateMonitoringSubscriptionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateMonitoringSubscriptionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateMonitoringSubscriptionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateMonitoringSubscriptionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlCreateMonitoringSubscriptionCommand(input, context);
    };
    CreateMonitoringSubscriptionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlCreateMonitoringSubscriptionCommand(output, context);
    };
    return CreateMonitoringSubscriptionCommand;
}($Command));
export { CreateMonitoringSubscriptionCommand };
