// @aws-sdk/client-cloudfront/dist-es/commands/CreateRealtimeLogConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateRealtimeLogConfigRequest, CreateRealtimeLogConfigResult } from "../models/models_0";
import { deserializeAws_restXmlCreateRealtimeLogConfigCommand, serializeAws_restXmlCreateRealtimeLogConfigCommand, } from "../protocols/Aws_restXml";
var CreateRealtimeLogConfigCommand = (function (_super) {
    __extends(CreateRealtimeLogConfigCommand, _super);
    function CreateRealtimeLogConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateRealtimeLogConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "CreateRealtimeLogConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateRealtimeLogConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateRealtimeLogConfigResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateRealtimeLogConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlCreateRealtimeLogConfigCommand(input, context);
    };
    CreateRealtimeLogConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlCreateRealtimeLogConfigCommand(output, context);
    };
    return CreateRealtimeLogConfigCommand;
}($Command));
export { CreateRealtimeLogConfigCommand };
