// @aws-sdk/client-cloudfront/dist-es/commands/CreateStreamingDistributionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateStreamingDistributionRequest, CreateStreamingDistributionResult } from "../models/models_0";
import { deserializeAws_restXmlCreateStreamingDistributionCommand, serializeAws_restXmlCreateStreamingDistributionCommand, } from "../protocols/Aws_restXml";
var CreateStreamingDistributionCommand = (function (_super) {
    __extends(CreateStreamingDistributionCommand, _super);
    function CreateStreamingDistributionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateStreamingDistributionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "CreateStreamingDistributionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateStreamingDistributionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateStreamingDistributionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateStreamingDistributionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlCreateStreamingDistributionCommand(input, context);
    };
    CreateStreamingDistributionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlCreateStreamingDistributionCommand(output, context);
    };
    return CreateStreamingDistributionCommand;
}($Command));
export { CreateStreamingDistributionCommand };
