// @aws-sdk/client-cloudfront/dist-es/commands/CreateCachePolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCachePolicyRequest, CreateCachePolicyResult } from "../models/models_0";
import { deserializeAws_restXmlCreateCachePolicyCommand, serializeAws_restXmlCreateCachePolicyCommand, } from "../protocols/Aws_restXml";
var CreateCachePolicyCommand = (function (_super) {
    __extends(CreateCachePolicyCommand, _super);
    function CreateCachePolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateCachePolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "CreateCachePolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCachePolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateCachePolicyResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCachePolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlCreateCachePolicyCommand(input, context);
    };
    CreateCachePolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlCreateCachePolicyCommand(output, context);
    };
    return CreateCachePolicyCommand;
}($Command));
export { CreateCachePolicyCommand };
