// @aws-sdk/client-cloudfront/dist-es/commands/CreateFieldLevelEncryptionProfileCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateFieldLevelEncryptionProfileRequest, CreateFieldLevelEncryptionProfileResult } from "../models/models_0";
import { deserializeAws_restXmlCreateFieldLevelEncryptionProfileCommand, serializeAws_restXmlCreateFieldLevelEncryptionProfileCommand, } from "../protocols/Aws_restXml";
var CreateFieldLevelEncryptionProfileCommand = (function (_super) {
    __extends(CreateFieldLevelEncryptionProfileCommand, _super);
    function CreateFieldLevelEncryptionProfileCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateFieldLevelEncryptionProfileCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "CreateFieldLevelEncryptionProfileCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateFieldLevelEncryptionProfileRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateFieldLevelEncryptionProfileResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateFieldLevelEncryptionProfileCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlCreateFieldLevelEncryptionProfileCommand(input, context);
    };
    CreateFieldLevelEncryptionProfileCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlCreateFieldLevelEncryptionProfileCommand(output, context);
    };
    return CreateFieldLevelEncryptionProfileCommand;
}($Command));
export { CreateFieldLevelEncryptionProfileCommand };
