// @aws-sdk/client-cloudfront/dist-es/commands/CreateOriginRequestPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateOriginRequestPolicyRequest, CreateOriginRequestPolicyResult } from "../models/models_0";
import { deserializeAws_restXmlCreateOriginRequestPolicyCommand, serializeAws_restXmlCreateOriginRequestPolicyCommand, } from "../protocols/Aws_restXml";
var CreateOriginRequestPolicyCommand = (function (_super) {
    __extends(CreateOriginRequestPolicyCommand, _super);
    function CreateOriginRequestPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateOriginRequestPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "CreateOriginRequestPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateOriginRequestPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateOriginRequestPolicyResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateOriginRequestPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlCreateOriginRequestPolicyCommand(input, context);
    };
    CreateOriginRequestPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlCreateOriginRequestPolicyCommand(output, context);
    };
    return CreateOriginRequestPolicyCommand;
}($Command));
export { CreateOriginRequestPolicyCommand };
