// @aws-sdk/client-cloudfront/dist-es/commands/DeleteFieldLevelEncryptionConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteFieldLevelEncryptionConfigRequest } from "../models/models_0";
import { deserializeAws_restXmlDeleteFieldLevelEncryptionConfigCommand, serializeAws_restXmlDeleteFieldLevelEncryptionConfigCommand, } from "../protocols/Aws_restXml";
var DeleteFieldLevelEncryptionConfigCommand = (function (_super) {
    __extends(DeleteFieldLevelEncryptionConfigCommand, _super);
    function DeleteFieldLevelEncryptionConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteFieldLevelEncryptionConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "DeleteFieldLevelEncryptionConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteFieldLevelEncryptionConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteFieldLevelEncryptionConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDeleteFieldLevelEncryptionConfigCommand(input, context);
    };
    DeleteFieldLevelEncryptionConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDeleteFieldLevelEncryptionConfigCommand(output, context);
    };
    return DeleteFieldLevelEncryptionConfigCommand;
}($Command));
export { DeleteFieldLevelEncryptionConfigCommand };
