// @aws-sdk/client-cloudfront/dist-es/commands/DeleteStreamingDistributionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteStreamingDistributionRequest } from "../models/models_1";
import { deserializeAws_restXmlDeleteStreamingDistributionCommand, serializeAws_restXmlDeleteStreamingDistributionCommand, } from "../protocols/Aws_restXml";
var DeleteStreamingDistributionCommand = (function (_super) {
    __extends(DeleteStreamingDistributionCommand, _super);
    function DeleteStreamingDistributionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteStreamingDistributionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "DeleteStreamingDistributionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteStreamingDistributionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteStreamingDistributionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDeleteStreamingDistributionCommand(input, context);
    };
    DeleteStreamingDistributionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDeleteStreamingDistributionCommand(output, context);
    };
    return DeleteStreamingDistributionCommand;
}($Command));
export { DeleteStreamingDistributionCommand };
