// @aws-sdk/client-cloudfront/dist-es/commands/DeleteKeyGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteKeyGroupRequest } from "../models/models_0";
import { deserializeAws_restXmlDeleteKeyGroupCommand, serializeAws_restXmlDeleteKeyGroupCommand, } from "../protocols/Aws_restXml";
var DeleteKeyGroupCommand = (function (_super) {
    __extends(DeleteKeyGroupCommand, _super);
    function DeleteKeyGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteKeyGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "DeleteKeyGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteKeyGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteKeyGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDeleteKeyGroupCommand(input, context);
    };
    DeleteKeyGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDeleteKeyGroupCommand(output, context);
    };
    return DeleteKeyGroupCommand;
}($Command));
export { DeleteKeyGroupCommand };
