// @aws-sdk/client-cloudfront/dist-es/commands/DeletePublicKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeletePublicKeyRequest } from "../models/models_1";
import { deserializeAws_restXmlDeletePublicKeyCommand, serializeAws_restXmlDeletePublicKeyCommand, } from "../protocols/Aws_restXml";
var DeletePublicKeyCommand = (function (_super) {
    __extends(DeletePublicKeyCommand, _super);
    function DeletePublicKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeletePublicKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "DeletePublicKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeletePublicKeyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeletePublicKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDeletePublicKeyCommand(input, context);
    };
    DeletePublicKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDeletePublicKeyCommand(output, context);
    };
    return DeletePublicKeyCommand;
}($Command));
export { DeletePublicKeyCommand };
