// @aws-sdk/client-cloudfront/dist-es/commands/DeleteCachePolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteCachePolicyRequest } from "../models/models_0";
import { deserializeAws_restXmlDeleteCachePolicyCommand, serializeAws_restXmlDeleteCachePolicyCommand, } from "../protocols/Aws_restXml";
var DeleteCachePolicyCommand = (function (_super) {
    __extends(DeleteCachePolicyCommand, _super);
    function DeleteCachePolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteCachePolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "DeleteCachePolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteCachePolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteCachePolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDeleteCachePolicyCommand(input, context);
    };
    DeleteCachePolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDeleteCachePolicyCommand(output, context);
    };
    return DeleteCachePolicyCommand;
}($Command));
export { DeleteCachePolicyCommand };
