// @aws-sdk/client-cloudfront/dist-es/commands/DeleteDistributionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteDistributionRequest } from "../models/models_0";
import { deserializeAws_restXmlDeleteDistributionCommand, serializeAws_restXmlDeleteDistributionCommand, } from "../protocols/Aws_restXml";
var DeleteDistributionCommand = (function (_super) {
    __extends(DeleteDistributionCommand, _super);
    function DeleteDistributionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDistributionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "DeleteDistributionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDistributionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDistributionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDeleteDistributionCommand(input, context);
    };
    DeleteDistributionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDeleteDistributionCommand(output, context);
    };
    return DeleteDistributionCommand;
}($Command));
export { DeleteDistributionCommand };
