// @aws-sdk/client-cloudfront/dist-es/commands/DeleteResponseHeadersPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteResponseHeadersPolicyRequest } from "../models/models_1";
import { deserializeAws_restXmlDeleteResponseHeadersPolicyCommand, serializeAws_restXmlDeleteResponseHeadersPolicyCommand, } from "../protocols/Aws_restXml";
var DeleteResponseHeadersPolicyCommand = (function (_super) {
    __extends(DeleteResponseHeadersPolicyCommand, _super);
    function DeleteResponseHeadersPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteResponseHeadersPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "DeleteResponseHeadersPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteResponseHeadersPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteResponseHeadersPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDeleteResponseHeadersPolicyCommand(input, context);
    };
    DeleteResponseHeadersPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDeleteResponseHeadersPolicyCommand(output, context);
    };
    return DeleteResponseHeadersPolicyCommand;
}($Command));
export { DeleteResponseHeadersPolicyCommand };
