// @aws-sdk/client-cloudfront/dist-es/commands/DescribeFunctionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeFunctionRequest, DescribeFunctionResult } from "../models/models_1";
import { deserializeAws_restXmlDescribeFunctionCommand, serializeAws_restXmlDescribeFunctionCommand, } from "../protocols/Aws_restXml";
var DescribeFunctionCommand = (function (_super) {
    __extends(DescribeFunctionCommand, _super);
    function DescribeFunctionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeFunctionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "DescribeFunctionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeFunctionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeFunctionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeFunctionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDescribeFunctionCommand(input, context);
    };
    DescribeFunctionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDescribeFunctionCommand(output, context);
    };
    return DescribeFunctionCommand;
}($Command));
export { DescribeFunctionCommand };
