// @aws-sdk/client-cloudfront/dist-es/commands/GetCloudFrontOriginAccessIdentityCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetCloudFrontOriginAccessIdentityRequest, GetCloudFrontOriginAccessIdentityResult } from "../models/models_1";
import { deserializeAws_restXmlGetCloudFrontOriginAccessIdentityCommand, serializeAws_restXmlGetCloudFrontOriginAccessIdentityCommand, } from "../protocols/Aws_restXml";
var GetCloudFrontOriginAccessIdentityCommand = (function (_super) {
    __extends(GetCloudFrontOriginAccessIdentityCommand, _super);
    function GetCloudFrontOriginAccessIdentityCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetCloudFrontOriginAccessIdentityCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "GetCloudFrontOriginAccessIdentityCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetCloudFrontOriginAccessIdentityRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetCloudFrontOriginAccessIdentityResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetCloudFrontOriginAccessIdentityCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetCloudFrontOriginAccessIdentityCommand(input, context);
    };
    GetCloudFrontOriginAccessIdentityCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetCloudFrontOriginAccessIdentityCommand(output, context);
    };
    return GetCloudFrontOriginAccessIdentityCommand;
}($Command));
export { GetCloudFrontOriginAccessIdentityCommand };
