// @aws-sdk/client-cloudfront/dist-es/commands/GetCachePolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetCachePolicyRequest, GetCachePolicyResult } from "../models/models_1";
import { deserializeAws_restXmlGetCachePolicyCommand, serializeAws_restXmlGetCachePolicyCommand, } from "../protocols/Aws_restXml";
var GetCachePolicyCommand = (function (_super) {
    __extends(GetCachePolicyCommand, _super);
    function GetCachePolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetCachePolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "GetCachePolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetCachePolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetCachePolicyResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetCachePolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetCachePolicyCommand(input, context);
    };
    GetCachePolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetCachePolicyCommand(output, context);
    };
    return GetCachePolicyCommand;
}($Command));
export { GetCachePolicyCommand };
