// @aws-sdk/client-cloudfront/dist-es/commands/GetFunctionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetFunctionRequest, GetFunctionResult } from "../models/models_1";
import { deserializeAws_restXmlGetFunctionCommand, serializeAws_restXmlGetFunctionCommand, } from "../protocols/Aws_restXml";
var GetFunctionCommand = (function (_super) {
    __extends(GetFunctionCommand, _super);
    function GetFunctionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetFunctionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "GetFunctionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetFunctionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetFunctionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetFunctionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetFunctionCommand(input, context);
    };
    GetFunctionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetFunctionCommand(output, context);
    };
    return GetFunctionCommand;
}($Command));
export { GetFunctionCommand };
