// @aws-sdk/client-cloudfront/dist-es/commands/GetFieldLevelEncryptionConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetFieldLevelEncryptionConfigRequest, GetFieldLevelEncryptionConfigResult } from "../models/models_1";
import { deserializeAws_restXmlGetFieldLevelEncryptionConfigCommand, serializeAws_restXmlGetFieldLevelEncryptionConfigCommand, } from "../protocols/Aws_restXml";
var GetFieldLevelEncryptionConfigCommand = (function (_super) {
    __extends(GetFieldLevelEncryptionConfigCommand, _super);
    function GetFieldLevelEncryptionConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetFieldLevelEncryptionConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "GetFieldLevelEncryptionConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetFieldLevelEncryptionConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetFieldLevelEncryptionConfigResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetFieldLevelEncryptionConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetFieldLevelEncryptionConfigCommand(input, context);
    };
    GetFieldLevelEncryptionConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetFieldLevelEncryptionConfigCommand(output, context);
    };
    return GetFieldLevelEncryptionConfigCommand;
}($Command));
export { GetFieldLevelEncryptionConfigCommand };
