// @aws-sdk/client-cloudfront/dist-es/commands/GetFieldLevelEncryptionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetFieldLevelEncryptionRequest, GetFieldLevelEncryptionResult } from "../models/models_1";
import { deserializeAws_restXmlGetFieldLevelEncryptionCommand, serializeAws_restXmlGetFieldLevelEncryptionCommand, } from "../protocols/Aws_restXml";
var GetFieldLevelEncryptionCommand = (function (_super) {
    __extends(GetFieldLevelEncryptionCommand, _super);
    function GetFieldLevelEncryptionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetFieldLevelEncryptionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "GetFieldLevelEncryptionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetFieldLevelEncryptionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetFieldLevelEncryptionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetFieldLevelEncryptionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetFieldLevelEncryptionCommand(input, context);
    };
    GetFieldLevelEncryptionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetFieldLevelEncryptionCommand(output, context);
    };
    return GetFieldLevelEncryptionCommand;
}($Command));
export { GetFieldLevelEncryptionCommand };
