// @aws-sdk/client-cloudfront/dist-es/commands/GetFieldLevelEncryptionProfileConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetFieldLevelEncryptionProfileConfigRequest, GetFieldLevelEncryptionProfileConfigResult, } from "../models/models_1";
import { deserializeAws_restXmlGetFieldLevelEncryptionProfileConfigCommand, serializeAws_restXmlGetFieldLevelEncryptionProfileConfigCommand, } from "../protocols/Aws_restXml";
var GetFieldLevelEncryptionProfileConfigCommand = (function (_super) {
    __extends(GetFieldLevelEncryptionProfileConfigCommand, _super);
    function GetFieldLevelEncryptionProfileConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetFieldLevelEncryptionProfileConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "GetFieldLevelEncryptionProfileConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetFieldLevelEncryptionProfileConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetFieldLevelEncryptionProfileConfigResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetFieldLevelEncryptionProfileConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetFieldLevelEncryptionProfileConfigCommand(input, context);
    };
    GetFieldLevelEncryptionProfileConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetFieldLevelEncryptionProfileConfigCommand(output, context);
    };
    return GetFieldLevelEncryptionProfileConfigCommand;
}($Command));
export { GetFieldLevelEncryptionProfileConfigCommand };
