// @aws-sdk/client-cloudfront/dist-es/commands/GetFieldLevelEncryptionProfileCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetFieldLevelEncryptionProfileRequest, GetFieldLevelEncryptionProfileResult } from "../models/models_1";
import { deserializeAws_restXmlGetFieldLevelEncryptionProfileCommand, serializeAws_restXmlGetFieldLevelEncryptionProfileCommand, } from "../protocols/Aws_restXml";
var GetFieldLevelEncryptionProfileCommand = (function (_super) {
    __extends(GetFieldLevelEncryptionProfileCommand, _super);
    function GetFieldLevelEncryptionProfileCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetFieldLevelEncryptionProfileCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "GetFieldLevelEncryptionProfileCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetFieldLevelEncryptionProfileRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetFieldLevelEncryptionProfileResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetFieldLevelEncryptionProfileCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetFieldLevelEncryptionProfileCommand(input, context);
    };
    GetFieldLevelEncryptionProfileCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetFieldLevelEncryptionProfileCommand(output, context);
    };
    return GetFieldLevelEncryptionProfileCommand;
}($Command));
export { GetFieldLevelEncryptionProfileCommand };
