// @aws-sdk/client-cloudfront/dist-es/commands/GetMonitoringSubscriptionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetMonitoringSubscriptionRequest, GetMonitoringSubscriptionResult } from "../models/models_1";
import { deserializeAws_restXmlGetMonitoringSubscriptionCommand, serializeAws_restXmlGetMonitoringSubscriptionCommand, } from "../protocols/Aws_restXml";
var GetMonitoringSubscriptionCommand = (function (_super) {
    __extends(GetMonitoringSubscriptionCommand, _super);
    function GetMonitoringSubscriptionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetMonitoringSubscriptionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "GetMonitoringSubscriptionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetMonitoringSubscriptionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetMonitoringSubscriptionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetMonitoringSubscriptionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetMonitoringSubscriptionCommand(input, context);
    };
    GetMonitoringSubscriptionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetMonitoringSubscriptionCommand(output, context);
    };
    return GetMonitoringSubscriptionCommand;
}($Command));
export { GetMonitoringSubscriptionCommand };
