// @aws-sdk/client-cloudfront/dist-es/commands/GetOriginRequestPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetOriginRequestPolicyRequest, GetOriginRequestPolicyResult } from "../models/models_1";
import { deserializeAws_restXmlGetOriginRequestPolicyCommand, serializeAws_restXmlGetOriginRequestPolicyCommand, } from "../protocols/Aws_restXml";
var GetOriginRequestPolicyCommand = (function (_super) {
    __extends(GetOriginRequestPolicyCommand, _super);
    function GetOriginRequestPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetOriginRequestPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "GetOriginRequestPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetOriginRequestPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetOriginRequestPolicyResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetOriginRequestPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetOriginRequestPolicyCommand(input, context);
    };
    GetOriginRequestPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetOriginRequestPolicyCommand(output, context);
    };
    return GetOriginRequestPolicyCommand;
}($Command));
export { GetOriginRequestPolicyCommand };
