// @aws-sdk/client-cloudfront/dist-es/commands/GetStreamingDistributionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetStreamingDistributionRequest, GetStreamingDistributionResult } from "../models/models_1";
import { deserializeAws_restXmlGetStreamingDistributionCommand, serializeAws_restXmlGetStreamingDistributionCommand, } from "../protocols/Aws_restXml";
var GetStreamingDistributionCommand = (function (_super) {
    __extends(GetStreamingDistributionCommand, _super);
    function GetStreamingDistributionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetStreamingDistributionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "GetStreamingDistributionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetStreamingDistributionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetStreamingDistributionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetStreamingDistributionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetStreamingDistributionCommand(input, context);
    };
    GetStreamingDistributionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetStreamingDistributionCommand(output, context);
    };
    return GetStreamingDistributionCommand;
}($Command));
export { GetStreamingDistributionCommand };
