// @aws-sdk/client-cloudfront/dist-es/commands/GetStreamingDistributionConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetStreamingDistributionConfigRequest, GetStreamingDistributionConfigResult } from "../models/models_1";
import { deserializeAws_restXmlGetStreamingDistributionConfigCommand, serializeAws_restXmlGetStreamingDistributionConfigCommand, } from "../protocols/Aws_restXml";
var GetStreamingDistributionConfigCommand = (function (_super) {
    __extends(GetStreamingDistributionConfigCommand, _super);
    function GetStreamingDistributionConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetStreamingDistributionConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "GetStreamingDistributionConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetStreamingDistributionConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetStreamingDistributionConfigResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetStreamingDistributionConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetStreamingDistributionConfigCommand(input, context);
    };
    GetStreamingDistributionConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetStreamingDistributionConfigCommand(output, context);
    };
    return GetStreamingDistributionConfigCommand;
}($Command));
export { GetStreamingDistributionConfigCommand };
