// @aws-sdk/client-cloudfront/dist-es/commands/ListFieldLevelEncryptionProfilesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListFieldLevelEncryptionProfilesRequest, ListFieldLevelEncryptionProfilesResult } from "../models/models_1";
import { deserializeAws_restXmlListFieldLevelEncryptionProfilesCommand, serializeAws_restXmlListFieldLevelEncryptionProfilesCommand, } from "../protocols/Aws_restXml";
var ListFieldLevelEncryptionProfilesCommand = (function (_super) {
    __extends(ListFieldLevelEncryptionProfilesCommand, _super);
    function ListFieldLevelEncryptionProfilesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListFieldLevelEncryptionProfilesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "ListFieldLevelEncryptionProfilesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListFieldLevelEncryptionProfilesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListFieldLevelEncryptionProfilesResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListFieldLevelEncryptionProfilesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListFieldLevelEncryptionProfilesCommand(input, context);
    };
    ListFieldLevelEncryptionProfilesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListFieldLevelEncryptionProfilesCommand(output, context);
    };
    return ListFieldLevelEncryptionProfilesCommand;
}($Command));
export { ListFieldLevelEncryptionProfilesCommand };
