// @aws-sdk/client-cloudfront/dist-es/commands/ListRealtimeLogConfigsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListRealtimeLogConfigsRequest, ListRealtimeLogConfigsResult } from "../models/models_1";
import { deserializeAws_restXmlListRealtimeLogConfigsCommand, serializeAws_restXmlListRealtimeLogConfigsCommand, } from "../protocols/Aws_restXml";
var ListRealtimeLogConfigsCommand = (function (_super) {
    __extends(ListRealtimeLogConfigsCommand, _super);
    function ListRealtimeLogConfigsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListRealtimeLogConfigsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "ListRealtimeLogConfigsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListRealtimeLogConfigsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListRealtimeLogConfigsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListRealtimeLogConfigsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListRealtimeLogConfigsCommand(input, context);
    };
    ListRealtimeLogConfigsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListRealtimeLogConfigsCommand(output, context);
    };
    return ListRealtimeLogConfigsCommand;
}($Command));
export { ListRealtimeLogConfigsCommand };
