// @aws-sdk/client-cloudfront/dist-es/commands/ListDistributionsByRealtimeLogConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListDistributionsByRealtimeLogConfigRequest, ListDistributionsByRealtimeLogConfigResult, } from "../models/models_1";
import { deserializeAws_restXmlListDistributionsByRealtimeLogConfigCommand, serializeAws_restXmlListDistributionsByRealtimeLogConfigCommand, } from "../protocols/Aws_restXml";
var ListDistributionsByRealtimeLogConfigCommand = (function (_super) {
    __extends(ListDistributionsByRealtimeLogConfigCommand, _super);
    function ListDistributionsByRealtimeLogConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListDistributionsByRealtimeLogConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "ListDistributionsByRealtimeLogConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListDistributionsByRealtimeLogConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListDistributionsByRealtimeLogConfigResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListDistributionsByRealtimeLogConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListDistributionsByRealtimeLogConfigCommand(input, context);
    };
    ListDistributionsByRealtimeLogConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListDistributionsByRealtimeLogConfigCommand(output, context);
    };
    return ListDistributionsByRealtimeLogConfigCommand;
}($Command));
export { ListDistributionsByRealtimeLogConfigCommand };
