// @aws-sdk/client-cloudfront/dist-es/commands/ListStreamingDistributionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListStreamingDistributionsRequest, ListStreamingDistributionsResult } from "../models/models_1";
import { deserializeAws_restXmlListStreamingDistributionsCommand, serializeAws_restXmlListStreamingDistributionsCommand, } from "../protocols/Aws_restXml";
var ListStreamingDistributionsCommand = (function (_super) {
    __extends(ListStreamingDistributionsCommand, _super);
    function ListStreamingDistributionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListStreamingDistributionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "ListStreamingDistributionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListStreamingDistributionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListStreamingDistributionsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListStreamingDistributionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListStreamingDistributionsCommand(input, context);
    };
    ListStreamingDistributionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListStreamingDistributionsCommand(output, context);
    };
    return ListStreamingDistributionsCommand;
}($Command));
export { ListStreamingDistributionsCommand };
