// @aws-sdk/client-cloudfront/dist-es/commands/ListDistributionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListDistributionsRequest, ListDistributionsResult } from "../models/models_1";
import { deserializeAws_restXmlListDistributionsCommand, serializeAws_restXmlListDistributionsCommand, } from "../protocols/Aws_restXml";
var ListDistributionsCommand = (function (_super) {
    __extends(ListDistributionsCommand, _super);
    function ListDistributionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListDistributionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "ListDistributionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListDistributionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListDistributionsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListDistributionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListDistributionsCommand(input, context);
    };
    ListDistributionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListDistributionsCommand(output, context);
    };
    return ListDistributionsCommand;
}($Command));
export { ListDistributionsCommand };
