// @aws-sdk/client-cloudfront/dist-es/commands/ListFunctionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListFunctionsRequest, ListFunctionsResult } from "../models/models_1";
import { deserializeAws_restXmlListFunctionsCommand, serializeAws_restXmlListFunctionsCommand, } from "../protocols/Aws_restXml";
var ListFunctionsCommand = (function (_super) {
    __extends(ListFunctionsCommand, _super);
    function ListFunctionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListFunctionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "ListFunctionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListFunctionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListFunctionsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListFunctionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListFunctionsCommand(input, context);
    };
    ListFunctionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListFunctionsCommand(output, context);
    };
    return ListFunctionsCommand;
}($Command));
export { ListFunctionsCommand };
