// @aws-sdk/client-cloudfront/dist-es/commands/ListConflictingAliasesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListConflictingAliasesRequest, ListConflictingAliasesResult } from "../models/models_1";
import { deserializeAws_restXmlListConflictingAliasesCommand, serializeAws_restXmlListConflictingAliasesCommand, } from "../protocols/Aws_restXml";
var ListConflictingAliasesCommand = (function (_super) {
    __extends(ListConflictingAliasesCommand, _super);
    function ListConflictingAliasesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListConflictingAliasesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "ListConflictingAliasesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListConflictingAliasesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListConflictingAliasesResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListConflictingAliasesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListConflictingAliasesCommand(input, context);
    };
    ListConflictingAliasesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListConflictingAliasesCommand(output, context);
    };
    return ListConflictingAliasesCommand;
}($Command));
export { ListConflictingAliasesCommand };
