// @aws-sdk/client-cloudfront/dist-es/commands/ListPublicKeysCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListPublicKeysRequest, ListPublicKeysResult } from "../models/models_1";
import { deserializeAws_restXmlListPublicKeysCommand, serializeAws_restXmlListPublicKeysCommand, } from "../protocols/Aws_restXml";
var ListPublicKeysCommand = (function (_super) {
    __extends(ListPublicKeysCommand, _super);
    function ListPublicKeysCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListPublicKeysCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "ListPublicKeysCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListPublicKeysRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListPublicKeysResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListPublicKeysCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListPublicKeysCommand(input, context);
    };
    ListPublicKeysCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListPublicKeysCommand(output, context);
    };
    return ListPublicKeysCommand;
}($Command));
export { ListPublicKeysCommand };
