// @aws-sdk/client-cloudfront/dist-es/commands/ListOriginRequestPoliciesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListOriginRequestPoliciesRequest, ListOriginRequestPoliciesResult } from "../models/models_1";
import { deserializeAws_restXmlListOriginRequestPoliciesCommand, serializeAws_restXmlListOriginRequestPoliciesCommand, } from "../protocols/Aws_restXml";
var ListOriginRequestPoliciesCommand = (function (_super) {
    __extends(ListOriginRequestPoliciesCommand, _super);
    function ListOriginRequestPoliciesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListOriginRequestPoliciesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "ListOriginRequestPoliciesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListOriginRequestPoliciesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListOriginRequestPoliciesResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListOriginRequestPoliciesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListOriginRequestPoliciesCommand(input, context);
    };
    ListOriginRequestPoliciesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListOriginRequestPoliciesCommand(output, context);
    };
    return ListOriginRequestPoliciesCommand;
}($Command));
export { ListOriginRequestPoliciesCommand };
