// @aws-sdk/client-cloudfront/dist-es/commands/TestFunctionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { TestFunctionRequest, TestFunctionResult } from "../models/models_1";
import { deserializeAws_restXmlTestFunctionCommand, serializeAws_restXmlTestFunctionCommand, } from "../protocols/Aws_restXml";
var TestFunctionCommand = (function (_super) {
    __extends(TestFunctionCommand, _super);
    function TestFunctionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    TestFunctionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "TestFunctionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: TestFunctionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: TestFunctionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    TestFunctionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlTestFunctionCommand(input, context);
    };
    TestFunctionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlTestFunctionCommand(output, context);
    };
    return TestFunctionCommand;
}($Command));
export { TestFunctionCommand };
