// @aws-sdk/client-cloudfront/dist-es/commands/UpdateResponseHeadersPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateResponseHeadersPolicyRequest, UpdateResponseHeadersPolicyResult } from "../models/models_1";
import { deserializeAws_restXmlUpdateResponseHeadersPolicyCommand, serializeAws_restXmlUpdateResponseHeadersPolicyCommand, } from "../protocols/Aws_restXml";
var UpdateResponseHeadersPolicyCommand = (function (_super) {
    __extends(UpdateResponseHeadersPolicyCommand, _super);
    function UpdateResponseHeadersPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateResponseHeadersPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "UpdateResponseHeadersPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateResponseHeadersPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateResponseHeadersPolicyResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateResponseHeadersPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlUpdateResponseHeadersPolicyCommand(input, context);
    };
    UpdateResponseHeadersPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlUpdateResponseHeadersPolicyCommand(output, context);
    };
    return UpdateResponseHeadersPolicyCommand;
}($Command));
export { UpdateResponseHeadersPolicyCommand };
