// @aws-sdk/client-cloudfront/dist-es/commands/UpdateStreamingDistributionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateStreamingDistributionRequest, UpdateStreamingDistributionResult } from "../models/models_1";
import { deserializeAws_restXmlUpdateStreamingDistributionCommand, serializeAws_restXmlUpdateStreamingDistributionCommand, } from "../protocols/Aws_restXml";
var UpdateStreamingDistributionCommand = (function (_super) {
    __extends(UpdateStreamingDistributionCommand, _super);
    function UpdateStreamingDistributionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateStreamingDistributionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "UpdateStreamingDistributionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateStreamingDistributionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateStreamingDistributionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateStreamingDistributionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlUpdateStreamingDistributionCommand(input, context);
    };
    UpdateStreamingDistributionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlUpdateStreamingDistributionCommand(output, context);
    };
    return UpdateStreamingDistributionCommand;
}($Command));
export { UpdateStreamingDistributionCommand };
