// @aws-sdk/client-cloudfront/dist-es/commands/UpdateFieldLevelEncryptionConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateFieldLevelEncryptionConfigRequest, UpdateFieldLevelEncryptionConfigResult } from "../models/models_1";
import { deserializeAws_restXmlUpdateFieldLevelEncryptionConfigCommand, serializeAws_restXmlUpdateFieldLevelEncryptionConfigCommand, } from "../protocols/Aws_restXml";
var UpdateFieldLevelEncryptionConfigCommand = (function (_super) {
    __extends(UpdateFieldLevelEncryptionConfigCommand, _super);
    function UpdateFieldLevelEncryptionConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateFieldLevelEncryptionConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "UpdateFieldLevelEncryptionConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateFieldLevelEncryptionConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateFieldLevelEncryptionConfigResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateFieldLevelEncryptionConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlUpdateFieldLevelEncryptionConfigCommand(input, context);
    };
    UpdateFieldLevelEncryptionConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlUpdateFieldLevelEncryptionConfigCommand(output, context);
    };
    return UpdateFieldLevelEncryptionConfigCommand;
}($Command));
export { UpdateFieldLevelEncryptionConfigCommand };
