// @aws-sdk/client-cloudfront/dist-es/commands/UpdateCachePolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateCachePolicyRequest, UpdateCachePolicyResult } from "../models/models_1";
import { deserializeAws_restXmlUpdateCachePolicyCommand, serializeAws_restXmlUpdateCachePolicyCommand, } from "../protocols/Aws_restXml";
var UpdateCachePolicyCommand = (function (_super) {
    __extends(UpdateCachePolicyCommand, _super);
    function UpdateCachePolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateCachePolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "UpdateCachePolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateCachePolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateCachePolicyResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateCachePolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlUpdateCachePolicyCommand(input, context);
    };
    UpdateCachePolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlUpdateCachePolicyCommand(output, context);
    };
    return UpdateCachePolicyCommand;
}($Command));
export { UpdateCachePolicyCommand };
