// @aws-sdk/client-cloudfront/dist-es/commands/UpdateFieldLevelEncryptionProfileCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateFieldLevelEncryptionProfileRequest, UpdateFieldLevelEncryptionProfileResult } from "../models/models_1";
import { deserializeAws_restXmlUpdateFieldLevelEncryptionProfileCommand, serializeAws_restXmlUpdateFieldLevelEncryptionProfileCommand, } from "../protocols/Aws_restXml";
var UpdateFieldLevelEncryptionProfileCommand = (function (_super) {
    __extends(UpdateFieldLevelEncryptionProfileCommand, _super);
    function UpdateFieldLevelEncryptionProfileCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateFieldLevelEncryptionProfileCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFrontClient";
        var commandName = "UpdateFieldLevelEncryptionProfileCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateFieldLevelEncryptionProfileRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateFieldLevelEncryptionProfileResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateFieldLevelEncryptionProfileCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlUpdateFieldLevelEncryptionProfileCommand(input, context);
    };
    UpdateFieldLevelEncryptionProfileCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlUpdateFieldLevelEncryptionProfileCommand(output, context);
    };
    return UpdateFieldLevelEncryptionProfileCommand;
}($Command));
export { UpdateFieldLevelEncryptionProfileCommand };
