// roamjs-components/components/ProgressDialog.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.render = void 0;
const tslib_1 = require("tslib");
const core_1 = require("@blueprintjs/core");
const react_1 = tslib_1.__importStar(require("react"));
const createOverlayRender_1 = tslib_1.__importDefault(require("../util/createOverlayRender"));
const ProgressDialog = ({ actionQueueLength: defaultActionQueueLength, timeout: defaultTimeout, onClose, }) => {
    const [actionQueueLength, setActionQueueLength] = (0, react_1.useState)(defaultActionQueueLength || 0);
    const [timeout, setTimeout] = (0, react_1.useState)(defaultTimeout || 60);
    (0, react_1.useEffect)(() => {
        const element = document.getElementById(`roamjs-progress-dialog-root`);
        if (element) {
            element.addEventListener("log", ((e) => {
                if (typeof e.detail.actionQueueLength !== "undefined")
                    setActionQueueLength(e.detail.actionQueueLength);
                if (typeof e.detail.timeout !== "undefined")
                    setTimeout(e.detail.timeout);
            }));
        }
        else {
            console.log(document.body);
            throw new Error(`Couldn't find roamjs-progress-dialog-root`);
        }
    }, [setActionQueueLength, setTimeout]);
    return (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement("style", null, `.roamjs-progress-dialog {
  right: 8px;
  margin: 0;
  position: absolute;
  bottom: 8px;
}
.roamjs-progress-dialog-portal .bp3-dialog-container,
.roamjs-progress-dialog-portal .bp3-overlay {
  pointer-events: none;
}
.roamjs-progress-dialog-portal .bp3-overlay-backdrop {
  display: none;
}`),
        react_1.default.createElement(core_1.Dialog, { isOpen: true, onClose: onClose, portalClassName: "roamjs-progress-dialog-portal", className: "roamjs-progress-dialog", canOutsideClickClose: false, canEscapeKeyClose: false, hasBackdrop: false, enforceFocus: false },
            react_1.default.createElement("div", { style: { padding: 16 } },
                react_1.default.createElement("h4", null, "Performing Write actions to Roam..."),
                react_1.default.createElement("p", null, "Roam only allows 300 writes per minute, waiting to finish."),
                react_1.default.createElement("p", null,
                    "Still have ",
                    react_1.default.createElement("b", null, actionQueueLength),
                    " actions to write in",
                    " ",
                    react_1.default.createElement("b", null, timeout),
                    " seconds...")))));
};
exports.render = (0, createOverlayRender_1.default)("progress-dialog", ProgressDialog);
exports.default = ProgressDialog;
